diff --git a/requirements.txt b/requirements.txt
index dd68eb2..a10e857 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -1,2 +1,3 @@
-future
-six
+# Requirements are listed in setup.py. The dot on the next line refers to
+# the current directory, instructing installers to use this package's setup.py
+.
diff --git a/setup.py b/setup.py
index 0d28fed..e266fcb 100644
--- a/setup.py
+++ b/setup.py
@@ -12,27 +12,13 @@ from __future__ import unicode_literals
 
 import io
 from os import path
-from pip.req import parse_requirements
 from setuptools import setup, find_packages
 
 
-def get_requirements(requirements_file):
-    """Use pip to parse requirements file."""
-    requirements = []
-    if path.isfile(requirements_file):
-        for req in parse_requirements(requirements_file, session="hack"):
-            # check markers, such as
-            #
-            #     rope_py3k    ; python_version >= '3.0'
-            #
-            if req.match_markers():
-                requirements.append(str(req.req))
-    return requirements
-
+INSTALL_REQUIRES = ['future', 'six']
 
 if __name__ == "__main__":
     HERE = path.abspath(path.dirname(__file__))
-    INSTALL_REQUIRES = get_requirements(path.join(HERE, "requirements.txt"))
     with io.open(path.join(HERE, "README.rst"), encoding="utf-8") as readme:
         LONG_DESCRIPTION = readme.read()
     setup(
