diff -u -r pango-1.43.0-orig/meson.build pango-1.43.0/meson.build
--- pango-1.43.0-orig/meson.build	2018-12-12 00:03:05.000000000 +0000
+++ pango-1.43.0/meson.build	2020-07-22 18:16:08.157214160 +0000
@@ -8,7 +8,8 @@
         ],
         meson_version : '>= 0.48.0')
 
-add_project_arguments([ '-D_POSIX_C_SOURCE=200809L', '-D_POSIX_THREAD_SAFE_FUNCTIONS', '-D_GNU_SOURCE', ], language: 'c')
+#add_project_arguments([ '-D_POSIX_C_SOURCE=200809L', '-D_POSIX_THREAD_SAFE_FUNCTIONS', '-D_GNU_SOURCE', ], language: 'c')
+add_project_arguments([ '-D_POSIX_THREAD_SAFE_FUNCTIONS', '-D_GNU_SOURCE', ], language: 'c')
 
 pango_prefix = get_option('prefix')
 pango_libdir = join_paths(pango_prefix, get_option('libdir'))
Only in pango-1.43.0: meson.build~
