diff -ru pip-10.0.1/docs/conf.py pip-10.0.1_patched/docs/conf.py
--- pip-10.0.1/docs/html/conf.py	2018-04-14 13:17:31.000000000 +0200
+++ pip-10.0.1_patched/docs/html/conf.py	2018-07-23 15:23:31.053267611 +0200
@@ -134,6 +134,7 @@
 # The theme to use for HTML and HTML Help pages.  Major themes that come with
 # Sphinx are currently 'default' and 'sphinxdoc'.
 html_theme = "pypa_theme"
+html_theme_path = ["pypa", "python-docs-theme"]
 
 # Theme options are theme-specific and customize the look and feel of a theme
 # further.  For a list of options available for each theme, see the
