diff -u -r xorgproto-2019.1-orig/include/X11/Xarch.h xorgproto-2019.1/include/X11/Xarch.h
--- xorgproto-2019.1-orig/include/X11/Xarch.h	2019-06-20 04:13:03.000000000 +0000
+++ xorgproto-2019.1/include/X11/Xarch.h	2020-10-27 17:54:16.313079200 +0000
@@ -41,7 +41,7 @@
 
 #  if defined(SVR4) || defined(__SVR4)
 #   include <sys/types.h>
-#   include <sys/byteorder.h>
+#   include <netman/protocols/byteorder.h>
 #  elif defined(CSRG_BASED)
 #   if defined(__NetBSD__) || defined(__OpenBSD__)
 #    include <sys/types.h>
diff -u -r xorgproto-2019.1-orig/include/X11/Xfuncs.h xorgproto-2019.1/include/X11/Xfuncs.h
--- xorgproto-2019.1-orig/include/X11/Xfuncs.h	2019-06-20 04:13:03.000000000 +0000
+++ xorgproto-2019.1/include/X11/Xfuncs.h	2020-10-27 17:54:45.621400800 +0000
@@ -37,14 +37,14 @@
 void bzero();
 int bcmp();
 #  else
-#   if defined(SYSV) && !defined(__SCO__) && !defined(__sun) && !defined(__UNIXWARE__) && !defined(_AIX)
+#   if defined(SYSV) && !defined(__SCO__) && !defined(__sun) && !defined(__UNIXWARE__) && !defined(_AIX) && !defined(__sgi)
 #    include <memory.h>
 void bcopy();
 #    define bzero(b,len) memset(b, 0, len)
 #    define bcmp(b1,b2,len) memcmp(b1, b2, len)
 #   else
 #    include <string.h>
-#    if defined(__SCO__) || defined(__sun) || defined(__UNIXWARE__) || defined(__CYGWIN__) || defined(_AIX) || defined(__APPLE__)
+#    if defined(__SCO__) || defined(__sun) || defined(__UNIXWARE__) || defined(__CYGWIN__) || defined(_AIX) || defined(__APPLE__) || defined(__sgi)
 #     include <strings.h>
 #    endif
 #    define _XFUNCS_H_INCLUDED_STRING_H
@@ -57,8 +57,11 @@
 #  ifndef _XFUNCS_H_INCLUDED_STRING_H
 #   include <string.h>
 #  endif
+
+# if !defined(__sgi)
 #  undef bzero
 #  define bzero(b,len) memset(b,0,len)
+# endif
 
 #  if defined WIN32 && defined __MINGW32__
 #   define bcopy(b1,b2,len) memmove(b2, b1, (size_t)(len))
diff -u -r xorgproto-2019.1-orig/include/X11/Xos.h xorgproto-2019.1/include/X11/Xos.h
--- xorgproto-2019.1-orig/include/X11/Xos.h	2019-06-20 04:13:03.000000000 +0000
+++ xorgproto-2019.1/include/X11/Xos.h	2020-10-27 17:54:16.315775520 +0000
@@ -60,7 +60,7 @@
  */
 
 # include <string.h>
-# if defined(__SCO__) || defined(__UNIXWARE__) || defined(__sun) || defined(__CYGWIN__) || defined(_AIX) || defined(__APPLE__)
+# if defined(__SCO__) || defined(__UNIXWARE__) || defined(__sun) || defined(__CYGWIN__) || defined(_AIX) || defined(__APPLE__) || defined(__sgi)
 #  include <strings.h>
 # else
 #  ifndef index
