diff --git a/tests/conftest.py b/tests/conftest.py
index ec47c86..174aebf 100644
--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -60,7 +60,7 @@ def serve():
             log.debug("shutting server down")
             server.shutdown()
             worker.join(1)
-            if worker.isAlive():
+            if worker.is_alive():
                 log.warning('worker is hanged')
             else:
                 log.debug("server stopped")
