diff -Naur libbonobo-2.32.1.orig/configure.in libbonobo-2.32.1/configure.in
--- libbonobo-2.32.1.orig/configure.in	2020-06-17 09:33:36.428079000 +0000
+++ libbonobo-2.32.1/configure.in	2020-06-17 10:18:50.763583151 +0000
@@ -188,7 +188,7 @@
 # glib-genmarshal
 AC_PATH_PROG(GLIB_GENMARSHAL, glib-genmarshal)
 
-GTK_DOC_CHECK([1.0])
+# GTK_DOC_CHECK([1.0])
 
 AC_ARG_ENABLE(debug,
 	      AS_HELP_STRING([--enable-debug],
@@ -269,11 +269,6 @@
 tests/Makefile
 tests/test-activation/Makefile
 tests/models/Makefile
-doc/Makefile
-doc/api/Makefile
-doc/api/figures/Makefile
-doc/activation/Makefile
-doc/activation-api/Makefile
 ])
 
 AC_OUTPUT
