diff -ur /usr/people/esp/rpmbuild/BUILD/Python-3.8.1/Lib/site.py Python-3.8.1/Lib/site.py
--- /usr/people/esp/rpmbuild/BUILD/Python-3.8.1/Lib/site.py	2019-12-18 13:21:23.000000000 +0000
+++ Python-3.8.1/Lib/site.py	2020-02-16 05:42:28.479435610 +0000
@@ -269,6 +269,10 @@
     if os.name == 'nt':
         return f'{userbase}\\Python{version[0]}{version[1]}\\site-packages'
 
+    # sgug-rse specific hack.
+    if sys.platform.startswith('irix'):
+        return f'{userbase}/lib32/python/site-packages'
+
     if sys.platform == 'darwin' and sys._framework:
         return f'{userbase}/lib/python/site-packages'
 
@@ -335,9 +339,15 @@
         seen.add(prefix)
 
         if os.sep == '/':
-            sitepackages.append(os.path.join(prefix, "lib",
-                                        "python%d.%d" % sys.version_info[:2],
-                                        "site-packages"))
+            # sgug-rse specific hack.
+            if sys.platform.startswith('irix'):
+                sitepackages.append(os.path.join(prefix, "lib32",
+                                                 "python%d.%d" % sys.version_info[:2],
+                                                 "site-packages"))
+            else:
+                sitepackages.append(os.path.join(prefix, "lib",
+                                                 "python%d.%d" % sys.version_info[:2],
+                                                 "site-packages"))
         else:
             sitepackages.append(prefix)
             sitepackages.append(os.path.join(prefix, "lib", "site-packages"))
diff -ur /usr/people/esp/rpmbuild/BUILD/Python-3.8.1/Lib/sysconfig.py Python-3.8.1/Lib/sysconfig.py
--- /usr/people/esp/rpmbuild/BUILD/Python-3.8.1/Lib/sysconfig.py	2019-12-18 13:21:23.000000000 +0000
+++ Python-3.8.1/Lib/sysconfig.py	2020-02-16 05:52:05.131899610 +0000
@@ -20,10 +20,10 @@
 
 _INSTALL_SCHEMES = {
     'posix_prefix': {
-        'stdlib': '{installed_base}/lib/python{py_version_short}',
-        'platstdlib': '{platbase}/lib/python{py_version_short}',
-        'purelib': '{base}/lib/python{py_version_short}/site-packages',
-        'platlib': '{platbase}/lib/python{py_version_short}/site-packages',
+        'stdlib': '{installed_base}/lib32/python{py_version_short}',
+        'platstdlib': '{platbase}/lib32/python{py_version_short}',
+        'purelib': '{base}/lib32/python{py_version_short}/site-packages',
+        'platlib': '{platbase}/lib32/python{py_version_short}/site-packages',
         'include':
             '{installed_base}/include/python{py_version_short}{abiflags}',
         'platinclude':
@@ -32,10 +32,10 @@
         'data': '{base}',
         },
     'posix_home': {
-        'stdlib': '{installed_base}/lib/python',
-        'platstdlib': '{base}/lib/python',
-        'purelib': '{base}/lib/python',
-        'platlib': '{base}/lib/python',
+        'stdlib': '{installed_base}/lib32/python',
+        'platstdlib': '{base}/lib32/python',
+        'purelib': '{base}/lib32/python',
+        'platlib': '{base}/lib32/python',
         'include': '{installed_base}/include/python',
         'platinclude': '{installed_base}/include/python',
         'scripts': '{base}/bin',
@@ -62,10 +62,10 @@
         'data': '{userbase}',
         },
     'posix_user': {
-        'stdlib': '{userbase}/lib/python{py_version_short}',
-        'platstdlib': '{userbase}/lib/python{py_version_short}',
-        'purelib': '{userbase}/lib/python{py_version_short}/site-packages',
-        'platlib': '{userbase}/lib/python{py_version_short}/site-packages',
+        'stdlib': '{userbase}/lib32/python{py_version_short}',
+        'platstdlib': '{userbase}/lib32/python{py_version_short}',
+        'purelib': '{userbase}/lib32/python{py_version_short}/site-packages',
+        'platlib': '{userbase}/lib32/python{py_version_short}/site-packages',
         'include': '{userbase}/include/python{py_version_short}',
         'scripts': '{userbase}/bin',
         'data': '{userbase}',
