diff -u -r libcanberra-0.30-orig/configure.ac libcanberra-0.30/configure.ac
--- libcanberra-0.30-orig/configure.ac	2012-09-25 01:21:07.000000000 +0000
+++ libcanberra-0.30/configure.ac	2020-11-28 12:59:48.160004760 +0000
@@ -114,10 +114,10 @@
         -fdata-sections])
 AC_SUBST([WARNINGFLAGS], $with_cflags)
 
-CC_CHECK_FLAGS_APPEND([with_ldflags], [LDFLAGS], [\
-        -Wl,--as-needed \
-        -Wl,--gc-sections])
-AC_SUBST([GCLDFLAGS], $with_ldflags)
+#CC_CHECK_FLAGS_APPEND([with_ldflags], [LDFLAGS], [\
+#        -Wl,--as-needed \
+#        -Wl,--gc-sections])
+#AC_SUBST([GCLDFLAGS], $with_ldflags)
 
 #### libtool stuff ####
 LT_PREREQ(2.2)
