diff -u -r libsolv-0.7.14-orig/src/repopage.c libsolv-0.7.14/src/repopage.c
--- libsolv-0.7.14-orig/src/repopage.c	2020-05-27 10:55:11.000000000 +0000
+++ libsolv-0.7.14/src/repopage.c	2020-07-27 19:22:22.017187120 +0000
@@ -18,6 +18,13 @@
  * pages automatically get dropped.
  */
 
+#if defined(__sgi)
+#include <unistd.h>
+#include <sys/types.h>
+#include <bstring.h>
+#include <sys/time.h>
+#endif
+
 #define _XOPEN_SOURCE 500
 
 #include <sys/types.h>
