diff -Naur gtk+-3.24.13.orig/docs/tools/shadow.c gtk+-3.24.13/docs/tools/shadow.c
--- gtk+-3.24.13.orig/docs/tools/shadow.c	2019-10-22 17:40:12.000000000 +0000
+++ gtk+-3.24.13/docs/tools/shadow.c	2020-11-24 08:14:49.426809000 +0000
@@ -1,5 +1,6 @@
 #include "shadow.h"
 #include <math.h>
+#include <values.h>
 
 #define BLUR_RADIUS 5
 #define SHADOW_OFFSET (BLUR_RADIUS * 4 / 5)
diff -Naur gtk+-3.24.13.orig/gtk/gtkcssimageradial.c gtk+-3.24.13/gtk/gtkcssimageradial.c
--- gtk+-3.24.13.orig/gtk/gtkcssimageradial.c	2019-11-27 10:23:51.000000000 +0000
+++ gtk+-3.24.13/gtk/gtkcssimageradial.c	2020-11-24 08:14:49.429309000 +0000
@@ -22,7 +22,7 @@
 #include "gtkcssimageradialprivate.h"
 
 #include <math.h>
-
+#include <values.h>
 #include "gtkcsscolorvalueprivate.h"
 #include "gtkcssnumbervalueprivate.h"
 #include "gtkcsspositionvalueprivate.h"
diff -Naur gtk+-3.24.13.orig/gtk/gtkprintoperation-portal.c gtk+-3.24.13/gtk/gtkprintoperation-portal.c
--- gtk+-3.24.13.orig/gtk/gtkprintoperation-portal.c	2019-11-27 10:23:51.000000000 +0000
+++ gtk+-3.24.13/gtk/gtkprintoperation-portal.c	2020-11-24 08:14:49.431430600 +0000
@@ -39,6 +39,9 @@
 #include "gtkwindowprivate.h"
 #include "gtkprivate.h"
 
+#ifdef __sgi
+#define O_CLOEXEC        02000000        /* set close_on_exec */
+#endif
 
 typedef struct {
   GtkPrintOperation *op;
diff -Naur gtk+-3.24.13.orig/gtk/gtkstylecontext.c gtk+-3.24.13/gtk/gtkstylecontext.c
--- gtk+-3.24.13.orig/gtk/gtkstylecontext.c	2019-11-27 10:23:51.000000000 +0000
+++ gtk+-3.24.13/gtk/gtkstylecontext.c	2020-11-24 09:15:36.546386600 +0000
@@ -1638,13 +1638,15 @@
 
   key.widget_type = widget_type;
   key.pspec = pspec;
-
+  
+  #if 0
   /* need value cache array */
   pcache = bsearch (&key,
                     priv->property_cache->data, priv->property_cache->len,
                     sizeof (PropertyValue), style_property_values_cmp);
   if (pcache)
     return &pcache->value;
+  #endif
 
   i = 0;
   while (i < priv->property_cache->len &&
diff -Naur gtk+-3.24.13.orig/tests/animated-resizing.c gtk+-3.24.13/tests/animated-resizing.c
--- gtk+-3.24.13.orig/tests/animated-resizing.c	2019-11-27 10:23:51.000000000 +0000
+++ gtk+-3.24.13/tests/animated-resizing.c	2020-11-24 08:14:49.449229800 +0000
@@ -2,7 +2,7 @@
 
 #include <gtk/gtk.h>
 #include <string.h>
-
+#include <values.h>
 #include "gtk/fallback-c89.c"
 #include "frame-stats.h"
 
diff -Naur gtk+-3.24.13.orig/tests/motion-compression.c gtk+-3.24.13/tests/motion-compression.c
--- gtk+-3.24.13.orig/tests/motion-compression.c	2019-11-27 10:23:51.000000000 +0000
+++ gtk+-3.24.13/tests/motion-compression.c	2020-11-24 08:14:49.450777000 +0000
@@ -1,6 +1,6 @@
 #include <gtk/gtk.h>
 #include <math.h>
-
+#include <values.h>
 GtkAdjustment *adjustment;
 int cursor_x, cursor_y;
 
