diff -rupN netsurf-all-3.10-orig/netsurf/Makefile netsurf-all-3.10/netsurf/Makefile
--- netsurf-all-3.10-orig/netsurf/Makefile	2020-05-24 14:35:22.000000000 +0000
+++ netsurf-all-3.10/netsurf/Makefile	2021-10-28 19:50:12.000000000 +0000
@@ -109,7 +109,7 @@ ifeq ($(TARGET),)
 endif
 
 # valid values for the TARGET
-VLDTARGET := riscos gtk2 gtk3 beos amiga amigaos3 framebuffer windows atari monkey
+VLDTARGET := riscos gtk2 gtk3 beos amiga amigaos3 framebuffer windows atari monkey motif
 
 # Check for valid TARGET
 ifeq ($(filter $(VLDTARGET),$(TARGET)),)
diff -rupN netsurf-all-3.10-orig/netsurf/Makefile.defaults netsurf-all-3.10/netsurf/Makefile.defaults
--- netsurf-all-3.10-orig/netsurf/Makefile.defaults	2020-05-24 14:35:22.000000000 +0000
+++ netsurf-all-3.10/netsurf/Makefile.defaults	2021-10-28 19:50:14.000000000 +0000
@@ -91,7 +91,8 @@ NETSURF_BUILTIN_VERBOSE_FILTER := "(leve
 
 # Force using glibc internal iconv implementation instead of external libiconv
 # Valid options: YES, NO
-NETSURF_USE_LIBICONV_PLUG := YES
+#NETSURF_USE_LIBICONV_PLUG := YES
+NETSURF_USE_LIBICONV_PLUG := NO
 
 # Enable use of utf8proc for international domain name processing
 # Valid options: YES, NO, AUTO	                          (highly recommended)
@@ -154,7 +155,6 @@ ifeq ($(HOST),FreeBSD)
     NETSURF_USE_HARU_PDF := NO
 endif
 
-
 # ----------------------------------------------------------------------------
 # Option overrides determined by specific TARGET
 # ----------------------------------------------------------------------------
diff -rupN netsurf-all-3.10-orig/netsurf/content/handlers/html/form.c netsurf-all-3.10/netsurf/content/handlers/html/form.c
--- netsurf-all-3.10-orig/netsurf/content/handlers/html/form.c	2020-05-24 14:35:22.000000000 +0000
+++ netsurf-all-3.10/netsurf/content/handlers/html/form.c	2021-10-28 19:50:15.000000000 +0000
@@ -727,7 +727,6 @@ form_dom_to_data_input_text(dom_html_inp
 				  charset,
 				  document_charset,
 				  fetch_data_next_ptr);
-
 	dom_string_unref(inputvalue);
 
 	return res;
@@ -803,7 +802,6 @@ form_dom_to_data_input(dom_html_input_el
 						    document_charset,
 						    submit_button,
 						    fetch_data_next_ptr);
-
 	} else if (dom_string_caseless_isequal(inputtype,
 					       corestring_dom_image)) {
 
@@ -1151,7 +1149,7 @@ form_dom_to_data(struct form *form,
 				form->document_charset,
 				&submit_button,
 				&fetch_data_next);
-
+ 
 		} else {
 			/* Form element is not handled */
 			NSLOG(netsurf, INFO,
diff -rupN netsurf-all-3.10-orig/netsurf/utils/nsoption.c netsurf-all-3.10/netsurf/utils/nsoption.c
--- netsurf-all-3.10-orig/netsurf/utils/nsoption.c	2020-05-24 14:35:23.000000000 +0000
+++ netsurf-all-3.10/netsurf/utils/nsoption.c	2021-10-28 19:50:12.000000000 +0000
@@ -67,6 +67,8 @@ static struct nsoption_s defaults[] = {
 #include "riscos/options.h"
 #elif defined(nsgtk)
 #include "gtk/options.h"
+#elif defined(nsmotif)
+#include "motif/options.h"
 #elif defined(nsbeos)
 #include "beos/options.h"
 #elif defined(nsamiga)
diff -rupN netsurf-all-3.10-orig/netsurf/utils/nsoption.h netsurf-all-3.10/netsurf/utils/nsoption.h
--- netsurf-all-3.10-orig/netsurf/utils/nsoption.h	2020-05-24 14:35:23.000000000 +0000
+++ netsurf-all-3.10/netsurf/utils/nsoption.h	2021-10-28 19:50:11.000000000 +0000
@@ -61,6 +61,8 @@
 #include "riscos/options.h"
 #elif defined(nsgtk)
 #include "gtk/options.h"
+#elif defined(nsmotif)
+#include "motif/options.h"
 #elif defined(nsbeos)
 #include "beos/options.h"
 #elif defined(nsamiga)
@@ -134,6 +136,8 @@ enum nsoption_e {
 #include "riscos/options.h"
 #elif defined(nsgtk)
 #include "gtk/options.h"
+#elif defined(nsmotif)
+#include "motif/options.h"
 #elif defined(nsbeos)
 #include "beos/options.h"
 #elif defined(nsamiga)
