diff -ru Sphinx-1.7.6/tests/test_theming.py Sphinx-1.7.6_patched/tests/test_theming.py
--- Sphinx-1.7.6/tests/test_theming.py	2018-07-16 11:24:40.000000000 +0200
+++ Sphinx-1.7.6_patched/tests/test_theming.py	2018-07-20 15:17:35.049263077 +0200
@@ -31,7 +31,7 @@
         themes.append('alabaster')
 
     # test Theme class API
-    assert set(app.html_themes.keys()) == set(themes)
+    assert set(app.html_themes.keys()) >= set(themes)
     assert app.html_themes['test-theme'] == app.srcdir / 'test_theme' / 'test-theme'
     assert app.html_themes['ziptheme'] == app.srcdir / 'ziptheme.zip'
     assert app.html_themes['staticfiles'] == app.srcdir / 'test_theme' / 'staticfiles'
