diff -u -r icu-orig/source/config/mh-irix icu/source/config/mh-irix
--- icu-orig/source/config/mh-irix	2019-04-11 23:38:30.000000000 +0000
+++ icu/source/config/mh-irix	2020-05-30 07:58:45.903438720 +0000
@@ -33,6 +33,9 @@
 LD_RPATH=
 LD_RPATH_PRE=  -Wl,-rpath,
 
+## Environment variable to set a runtime search path
+LDLIBRARYPATH_ENVVAR = LD_LIBRARYN32_PATH
+
 ## Compiler switch to embed a library name
 LD_SONAME = -Wl,-soname -Wl,$(notdir $(MIDDLE_SO_TARGET))
 
diff -u -r icu-orig/source/i18n/rbnf.cpp icu/source/i18n/rbnf.cpp
--- icu-orig/source/i18n/rbnf.cpp	2019-04-11 23:38:30.000000000 +0000
+++ icu/source/i18n/rbnf.cpp	2020-05-30 07:59:15.484235400 +0000
@@ -289,7 +289,7 @@
     OPEN_ANGLE = 0x003c, /* '<' */
     CLOSE_ANGLE = 0x003e, /* '>' */
     COMMA = 0x002c,
-    TICK = 0x0027,
+    ICU_TICK = 0x0027,
     QUOTE = 0x0022,
     SPACE = 0x0020
 };
@@ -368,11 +368,11 @@
 };
 
 static const UChar SQUOTE_STOPLIST[] = { 
-    TICK, 0
+    ICU_TICK, 0
 };
 
 static const UChar NOQUOTE_STOPLIST[] = { 
-    SPACE, COMMA, CLOSE_ANGLE, OPEN_ANGLE, TICK, QUOTE, 0
+    SPACE, COMMA, CLOSE_ANGLE, OPEN_ANGLE, ICU_TICK, QUOTE, 0
 };
 
 static void
@@ -522,7 +522,7 @@
     if (p < e) {
         const UChar* terminators;
         UChar c = *p;
-        UBool haveQuote = c == QUOTE || c == TICK;
+        UBool haveQuote = c == QUOTE || c == ICU_TICK;
         if (haveQuote) {
             inc();
             terminators = c == QUOTE ? DQUOTE_STOPLIST : SQUOTE_STOPLIST;
@@ -548,7 +548,7 @@
                 ERROR("Empty string");
             }
             inc();
-        } else if (x == OPEN_ANGLE || x == TICK || x == QUOTE) {
+        } else if (x == OPEN_ANGLE || x == ICU_TICK || x == QUOTE) {
             ERROR("Unexpected character in string");
         }
     }
