--- a/setup.py	2015-11-09 23:38:34.362909618 +0100
+++ b/setup.py	2015-11-09 23:38:56.425158723 +0100
@@ -6,9 +6,9 @@
 install_requires = [
     'xlrd>=0.7.1',
     'sqlalchemy>=0.6.6',
-    'openpyxl==2.2.0-b1',
+    'openpyxl>=2.2.0-b1',
     'six>=1.6.1',
-    'python-dateutil==2.2'
+    'python-dateutil>=2.2'
 ]

 if sys.version_info < (2, 7):
@@ -17,7 +17,7 @@
     install_requires.append('simplejson>=3.6.3')

 if sys.version_info[0] == 2:
-    install_requires.append('dbf==0.94.003')
+    install_requires.append('dbf>=0.94.003')

 setup(
     name='csvkit',
