diff --git i/csvkit/convert/xlsx.py w/csvkit/convert/xlsx.py
index 6f8c23e..2ca466d 100644
--- i/csvkit/convert/xlsx.py
+++ w/csvkit/convert/xlsx.py
@@ -48,7 +48,7 @@ def xlsx2csv(f, output=None, **kwargs):
 
     writer = CSVKitWriter(output)
 
-    book = load_workbook(f, use_iterators=True, data_only=True)
+    book = load_workbook(f, data_only=True)
 
     if 'sheet' in kwargs:
         sheet = book.get_sheet_by_name(kwargs['sheet'])
