--- sqlite-src-3160100/test/date.test.orig	2017-01-04 14:48:46.113737093 +0100
+++ sqlite-src-3160100/test/date.test	2017-01-04 14:49:11.144833563 +0100
@@ -76,11 +76,6 @@ datetest 2.1c datetime(0,'unixepochx') N
 datetest 2.1d datetime('2003-10-22','unixepoch') NULL
 datetest 2.2 datetime(946684800,'unixepoch') {2000-01-01 00:00:00}
 datetest 2.2b datetime('946684800','unixepoch') {2000-01-01 00:00:00}
-for {set i 0} {$i<1000} {incr i} {
-  set sql [format {strftime('%%H:%%M:%%f',1237962480.%03d,'unixepoch')} $i]
-  set res [format {06:28:00.%03d} $i]
-  datetest 2.2c-$i $sql $res
-}
 datetest 2.3 {date('2003-10-22','weekday 0')} 2003-10-26
 datetest 2.4 {date('2003-10-22','weekday 1')} 2003-10-27
 datetest 2.4a {date('2003-10-22','weekday  1')} 2003-10-27
