--- trollius/tasks.py.orig	2016-02-05 05:14:29.000000000 -0700
+++ trollius/tasks.py	2018-07-21 14:20:25.014870895 -0600
@@ -3,7 +3,7 @@ from __future__ import print_function
 
 __all__ = ['Task',
            'FIRST_COMPLETED', 'FIRST_EXCEPTION', 'ALL_COMPLETED',
-           'wait', 'wait_for', 'as_completed', 'sleep', 'async',
+           'wait', 'wait_for', 'as_completed', 'sleep', 'async_',
            'gather', 'shield', 'ensure_future',
            ]
 
@@ -562,7 +562,7 @@ def sleep(delay, result=None, loop=None)
         h.cancel()
 
 
-def async(coro_or_future, loop=None):
+def async_(coro_or_future, loop=None):
     """Wrap a coroutine in a future.
 
     If the argument is a Future, it is returned directly.
