diff -Naur devilutionX-1.0.orig/SourceS/file_util.h devilutionX-1.0/SourceS/file_util.h
--- devilutionX-1.0.orig/SourceS/file_util.h	2021-02-14 09:08:43.000000000 +0000
+++ devilutionX-1.0/SourceS/file_util.h	2021-02-14 15:17:44.042986000 +0000
@@ -16,7 +16,7 @@
 #include <windows.h>
 #endif
 
-#if _POSIX_C_SOURCE >= 200112L || defined(_BSD_SOURCE) || defined(__APPLE__)
+#if _POSIX_C_SOURCE >= 200112L || defined(_BSD_SOURCE) || defined(__APPLE__) || defined(_SGI_SOURCE)
 #include <unistd.h>
 #include <sys/stat.h>
 #else
@@ -27,7 +27,7 @@
 
 inline bool FileExists(const char *path)
 {
-#if _POSIX_C_SOURCE >= 200112L || defined(_BSD_SOURCE) || defined(__APPLE__)
+#if _POSIX_C_SOURCE >= 200112L || defined(_BSD_SOURCE) || defined(__APPLE__) || defined(_SGI_SOURCE)
 	return ::access(path, F_OK) == 0;
 #else
 	FILE *file = std::fopen(path, "rb");
@@ -87,7 +87,7 @@
 	}
 	::CloseHandle(file);
 	return true;
-#elif _POSIX_C_SOURCE >= 200112L || defined(_BSD_SOURCE) || defined(__APPLE__)
+#elif _POSIX_C_SOURCE >= 200112L || defined(_BSD_SOURCE) || defined(__APPLE__) || defined(_SGI_SOURCE)
 	return ::truncate(path, static_cast<off_t>(size)) == 0;
 #else
 	static_assert(false, "truncate not implemented for the current platform");
diff -Naur devilutionX-1.0.orig/SourceS/sdl2_to_1_2_backports.h devilutionX-1.0/SourceS/sdl2_to_1_2_backports.h
--- devilutionX-1.0.orig/SourceS/sdl2_to_1_2_backports.h	2021-02-14 09:08:43.000000000 +0000
+++ devilutionX-1.0/SourceS/sdl2_to_1_2_backports.h	2021-02-14 14:27:39.072186560 +0000
@@ -1,13 +1,14 @@
 #pragma once
 
 #include <SDL.h>
-#include <unistd.h>
 #include <errno.h>
+#include <stdlib.h>
 #include <stdio.h>
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <math.h>
 #include <cstddef>
+#include <unistd.h>
 
 #include "../SourceX/stubs.h"
 
@@ -678,13 +679,13 @@
 //== Filesystem
 
 #if !defined(__QNXNTO__)
-inline char *
-readSymLink(const char *path)
+inline char * readSymLink(const char *path)
 {
 	// From sdl2-2.0.9/src/filesystem/unix/SDL_sysfilesystem.c
 	char *retval = NULL;
 	ssize_t len = 64;
-	ssize_t rc = -1;
+	// ssize_t rc = -1;
+	int rc = -1;
 
 	while (1) {
 		char *ptr = (char *)SDL_realloc(retval, (size_t)len);
diff -Naur devilutionX-1.0.orig/build/CMakeLists.txt devilutionX-1.0/build/CMakeLists.txt
--- devilutionX-1.0.orig/build/CMakeLists.txt	2021-02-14 09:23:18.000000000 +0000
+++ devilutionX-1.0/build/CMakeLists.txt	2021-02-14 15:05:56.461748560 +0000
@@ -527,10 +527,10 @@
   endif()
 endif()
 
-if(NOT WIN32 AND NOT APPLE AND NOT ${CMAKE_SYSTEM_NAME} STREQUAL FreeBSD)
+# if(NOT WIN32 AND NOT APPLE AND NOT ${CMAKE_SYSTEM_NAME} STREQUAL FreeBSD)
   # Enable POSIX extensions such as `readlink` and `ftruncate`.
-  add_definitions(-D_POSIX_C_SOURCE=200809L)
-endif()
+  #  add_definitions(-D_POSIX_C_SOURCE=200809L)
+#  endif()
 
 if(HAIKU)
   target_link_libraries(${BIN_TARGET} PRIVATE network)
