diff -uNr cssutils-1.0.1.orig/setup.py cssutils-1.0.1/setup.py
--- cssutils-1.0.1.orig/setup.py	2015-10-11 07:37:34.000000000 +0200
+++ cssutils-1.0.1/setup.py	2016-08-27 10:30:27.077897978 +0200
@@ -50,7 +50,7 @@
     package_dir={'':'src'},
     packages=find_packages('src'),
     test_suite='cssutils.tests', #'nose.collector'
-    tests_require=['mock', 'pbr < 1.7.0'],
+    tests_require=['mock'],
     entry_points={
         'console_scripts': [
             'csscapture = cssutils.scripts.csscapture:main',
