Index: pyside-setup-opensource-src-5.14.2/sources/pyside2-tools/CMakeLists.txt
===================================================================
--- pyside-setup-opensource-src-5.14.2.orig/sources/pyside2-tools/CMakeLists.txt
+++ pyside-setup-opensource-src-5.14.2/sources/pyside2-tools/CMakeLists.txt
@@ -47,28 +47,12 @@ else()
 endif()
 
 set(TOOLS_PATH "${_qt5Core_install_prefix}/bin")
-set(UIC_PATH "${TOOLS_PATH}/uic${EXE_EXT}")
-set(RCC_PATH "${TOOLS_PATH}/rcc${EXE_EXT}")
 if (APPLE)
     set(DESIGNER_PATH "${TOOLS_PATH}/Designer.app")
 else()
     set(DESIGNER_PATH "${TOOLS_PATH}/designer${EXE_EXT}")
 endif()
 
-install(FILES "${UIC_PATH}"
-        DESTINATION bin
-        PERMISSIONS
-        OWNER_EXECUTE OWNER_WRITE OWNER_READ
-        GROUP_EXECUTE GROUP_READ
-        WORLD_EXECUTE WORLD_READ)
-
-install(FILES "${RCC_PATH}"
-        DESTINATION bin
-        PERMISSIONS
-        OWNER_EXECUTE OWNER_WRITE OWNER_READ
-        GROUP_EXECUTE GROUP_READ
-        WORLD_EXECUTE WORLD_READ)
-
 if (EXISTS ${DESIGNER_PATH})
     if (APPLE)
         install(DIRECTORY "${DESIGNER_PATH}"
