diff -up ./setup.py.orig ./setup.py
--- ./setup.py.orig	2016-06-22 21:26:16.000000000 +0200
+++ ./setup.py	2017-07-02 22:17:54.301963250 +0200
@@ -83,7 +83,7 @@ else:
     # Force `setup_requires` stuff like Cython to be installed before proceeding
     #
     from setuptools.dist import Distribution
-    Distribution(dict(setup_requires='Cython>=0.19.1'))
+    Distribution(dict(setup_requires='Cython>=0.19'))
 
     from Cython.Distutils import build_ext as _build_ext
 from distutils.dir_util import remove_tree
