diff --git a/CMakeLists.txt b/CMakeLists.txt
index 18b242db..b7694414 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -342,7 +342,7 @@ elseif (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
         )
     endif()
 
-elseif (${CMAKE_SYSTEM_NAME} MATCHES "Linux")
+elseif (${CMAKE_SYSTEM_NAME} MATCHES "Linux|IRIX*")
     set (OPENSSL_LIBS ssl crypto)
 else()
     find_library (lib_ssl ssl)
@@ -418,7 +418,7 @@ endif()
 #
 # Linux installation
 #
-if (${CMAKE_SYSTEM_NAME} MATCHES "Linux")
+if (${CMAKE_SYSTEM_NAME} MATCHES "Linux|IRIX*")
     configure_files (${CMAKE_CURRENT_SOURCE_DIR}/dist/rpm ${CMAKE_BINARY_DIR}/rpm)
     install(FILES res/barrier.svg DESTINATION share/icons/hicolor/scalable/apps)
     if("${VERSION_MAJOR}" STREQUAL "2") 
