--- fltk-1.3.5/src/fl_read_image.cxx.orig	2020-04-19 21:33:18.481544080 +0000
+++ fltk-1.3.5/src/fl_read_image.cxx	2020-04-19 21:34:09.023938560 +0000
@@ -246,7 +246,7 @@
   int allow_outside = w < 0;    // negative w allows negative X or Y, that is, window frame
   if (w < 0) w = - w;
 
-#  ifdef __sgi
+#if defined(__sgi) && !defined(__GNUC__)
   if (XReadDisplayQueryExtension(fl_display, &i, &i)) {
     image = XReadDisplay(fl_display, fl_window, X, Y, w, h, 0, NULL);
   } else
