--- tdnf-3.1.5-orig/client/defines.h	2021-10-06 18:39:26.000000000 +0000
+++ tdnf-3.1.5/client/defines.h	2021-11-17 20:39:21.042920320 +0000
@@ -59,7 +59,7 @@
 
 //Repo defines
 #define TDNF_REPO_EXT                     ".repo"
-#define TDNF_CONF_FILE                    "/etc/tdnf/tdnf.conf"
+#define TDNF_CONF_FILE                    "/usr/sgug/etc/tdnf/tdnf.conf"
 #define TDNF_CONF_GROUP                   "main"
 //Conf file key names
 #define TDNF_CONF_KEY_GPGCHECK            "gpgcheck"
@@ -110,10 +110,10 @@
 #define TDNF_REPO_BASEURL_FILE_NAME       "baseurl"
 
 // repo defaults
-#define TDNF_DEFAULT_REPO_LOCATION        "/etc/yum.repos.d"
-#define TDNF_DEFAULT_CACHE_LOCATION       "/var/cache/tdnf"
-#define TDNF_DEFAULT_DISTROVERPKG         "system-release"
-#define TDNF_DEFAULT_DISTROARCHPKG        "x86_64"
+#define TDNF_DEFAULT_REPO_LOCATION        "/usr/sgug/etc/yum.repos.d"
+#define TDNF_DEFAULT_CACHE_LOCATION       "/usr/sgug/var/cache/tdnf"
+#define TDNF_DEFAULT_DISTROVERPKG         "sgugrse-release"
+#define TDNF_DEFAULT_DISTROARCHPKG        "mips"
 #define TDNF_RPM_CACHE_DIR_NAME           "rpms"
 #define TDNF_REPODATA_DIR_NAME            "repodata"
 #define TDNF_SOLVCACHE_DIR_NAME           "solvcache"
@@ -140,7 +140,7 @@
 /* plugin defines */
 #define TDNF_DEFAULT_PLUGINS_ENABLED      0
 #define TDNF_DEFAULT_PLUGIN_PATH          SYSTEM_LIBDIR"/tdnf-plugins"
-#define TDNF_DEFAULT_PLUGIN_CONF_PATH     "/etc/tdnf/pluginconf.d"
+#define TDNF_DEFAULT_PLUGIN_CONF_PATH     "/usr/sgug/etc/tdnf/pluginconf.d"
 #define TDNF_PLUGIN_CONF_EXT              ".conf"
 #define TDNF_PLUGIN_CONF_EXT_LEN          5
 #define TDNF_PLUGIN_CONF_MAIN_SECTION     "main"
