diff -ru techtonik-hexdump-66325cb5fed8.orig/hexdump.py techtonik-hexdump-66325cb5fed8.new/hexdump.py
--- techtonik-hexdump-66325cb5fed8.orig/hexdump.py	2018-01-04 23:00:28.831522944 +0100
+++ techtonik-hexdump-66325cb5fed8.new/hexdump.py	2018-01-04 23:00:58.018100187 +0100
@@ -23,7 +23,7 @@
 
 """
 
-__version__ = '3.4dev'
+__version__ = '3.4'
 __author__  = 'anatoly techtonik <techtonik@gmail.com>'
 __license__ = 'Public Domain'
 
Only in techtonik-hexdump-66325cb5fed8.new: hexdump.py.setup
diff -ru techtonik-hexdump-66325cb5fed8.orig/setup.py techtonik-hexdump-66325cb5fed8.new/setup.py
--- techtonik-hexdump-66325cb5fed8.orig/setup.py	2016-08-18 09:36:52.000000000 +0200
+++ techtonik-hexdump-66325cb5fed8.new/setup.py	2018-01-04 23:10:29.006829691 +0100
@@ -43,7 +43,12 @@
 import os
 import textwrap
 import zipfile
-from distutils.core import setup
+
+try:
+    from setuptools import setup
+except ImportError:
+    from distutils.core import setup
+
 from distutils import log
 from distutils import archive_util
 
@@ -91,29 +96,44 @@
 
 # /-- distutils config
 
-setup(
-    name='hexdump',
-    version=get_version('hexdump.py'),
-    author='anatoly techtonik <techtonik@gmail.com>',
-    url='https://bitbucket.org/techtonik/hexdump/',
-
-    description="view/edit your binary with any text editor",
-    license="Public Domain",
-    classifiers=[
-        #'Environment :: Console',
-        'License :: Public Domain',
-        'Operating System :: OS Independent',
-        'Programming Language :: Python :: 2',
-        'Programming Language :: Python :: 3',
-        'Topic :: Software Development',
-        'Topic :: Software Development :: Libraries :: Python Modules',
-        #'Topic :: Utilities',
-    ],
+def main():
 
-    py_modules=['hexdump'],
-    data_files=[('data', ['data/hexfile.bin'])],
+    dist = setup(
+        name='hexdump',
+        version=get_version('hexdump.py'),
+        author='anatoly techtonik',
+        author_email='techtonik@gmail.com',
+        url='https://bitbucket.org/techtonik/hexdump/',
+
+        description="view/edit your binary with any text editor",
+        license="Public Domain",
+        classifiers=[
+            #'Environment :: Console',
+            'License :: Public Domain',
+            'Operating System :: OS Independent',
+            'Programming Language :: Python :: 2',
+            'Programming Language :: Python :: 3',
+            'Topic :: Software Development',
+            'Topic :: Software Development :: Libraries :: Python Modules',
+            #'Topic :: Utilities',
+        ],
+
+        packages=['hexdump'],
+        package_dir={'hexdump': '.'},
+        package_data={'hexdump': ['data/hexfile.bin']},
+
+        long_description=get_description('README.txt'),
+
+        cmdclass={'sdist': sdistzip},
+
+        entry_points={
+            'console_scripts': [
+                'hexdumpy = hexdump:main',
+            ]
+        },
+)
 
-    long_description=get_description('README.txt'),
 
-    cmdclass={'sdist': sdistzip},
-)
+if __name__ == "__main__":
+    main()
+
Only in techtonik-hexdump-66325cb5fed8.new: setup.py.setup
