diff --git a/test.py b/test.py
index 27c8faf..dffeaba 100644
--- a/test.py
+++ b/test.py
@@ -1975,7 +1975,7 @@ sys.stdout.write("line1")
         try:
             sh.sleep(sleep_for, _timeout=timeout).wait()
         except sh.TimeoutException as e:
-            self.assertEqual(e.full_cmd, '/bin/sleep 3')
+            self.assertEqual(e.full_cmd, '/usr/bin/sleep 3')
         else:
             self.fail("no timeout exception")
         elapsed = time() - started
