diff -ur tar-1.30/tests/checkseekhole.c tar-1.30-patched/tests/checkseekhole.c
--- tar-1.30/tests/checkseekhole.c	Mon Jan 02 12:43:50 2017
+++ tar-1.30-patched/tests/checkseekhole.c	Tue Jun 05 15:43:37 2018
@@ -27,9 +27,9 @@
 #include <fcntl.h>
 
 enum {
-    EX_OK = 0,    /* SEEK_HOLE support */
-    EX_FAIL,      /* test failed - no SEEK_HOLE support */
-    EX_BAD,       /* test is not relevant */
+    MY_EX_OK = 0,    /* SEEK_HOLE support */
+    MY_EX_FAIL,      /* test failed - no SEEK_HOLE support */
+    MY_EX_BAD,       /* test is not relevant */
 };
 
 int
@@ -41,34 +41,34 @@
 
   /* hole of 100MB */
   if (lseek (fd, 100*1024*1024, SEEK_END) < 0)
-    return EX_BAD;
+    return MY_EX_BAD;
 
   /* piece of data */
   if (write (fd, "data\n", 5) != 5)
-    return EX_BAD;
+    return MY_EX_BAD;
 
   /* another hole */
   if (lseek (fd, 100*1024*1024, SEEK_END) < 0)
-    return EX_BAD;
+    return MY_EX_BAD;
 
   /* piece of data */
   if (write (fd, "data\n", 5) != 5)
-    return EX_BAD;
+    return MY_EX_BAD;
 
   if (fstat (fd, &stat))
-    return EX_BAD;
+    return MY_EX_BAD;
 
   offset = lseek (fd, 0, SEEK_DATA);
   if (offset == (off_t)-1)
-    return EX_FAIL;
+    return MY_EX_FAIL;
 
   offset = lseek (fd, offset, SEEK_HOLE);
   if (offset == (off_t)-1 || offset == stat.st_size)
-    return EX_FAIL;
+    return MY_EX_FAIL;
 
-  return EX_OK;
+  return MY_EX_OK;
 #else
-  return EX_BAD;
+  return MY_EX_BAD;
 #endif
 }
 
@@ -80,7 +80,7 @@
   char template[] = "testseekhole-XXXXXX";
   int fd = mkstemp (template);
   if (fd == -1)
-    return EX_BAD;
+    return MY_EX_BAD;
   rc = check_seek_hole (fd);
   close (fd);
   unlink (template);
@@ -87,6 +87,6 @@
 
   return rc;
 #else
-  return EX_FAIL;
+  return MY_EX_FAIL;
 #endif
 }
