--- rpm-4.15.0/macros.in~	2020-01-24 10:38:50.947048240 +0000
+++ rpm-4.15.0/macros.in	2020-01-26 06:45:59.805630800 +0000
@@ -812,6 +812,8 @@
   export RPM_BUILD_ROOT}\
   %{?_javaclasspath:CLASSPATH=\"%{_javaclasspath}\"\
   export CLASSPATH}\
+  LD_LIBRARYN32_PATH=\"%{_libdir}:/usr/lib32:/lib32"\
+  export LD_LIBRARYN32_PATH\
   PKG_CONFIG_PATH=\"${PKG_CONFIG_PATH}:%{_libdir}/pkgconfig:%{_datadir}/pkgconfig\"\
   export PKG_CONFIG_PATH\
   CONFIG_SITE=${CONFIG_SITE:-NONE}\
