diff -N -r -u -w rpm-4.15.0/lib/rpmfi.c rpm-4.15.0-patched/lib/rpmfi.c
--- rpm-4.15.0/lib/rpmfi.c	2019-06-26 15:17:31.413985000 +0000
+++ rpm-4.15.0-patched/lib/rpmfi.c	2019-10-25 17:28:11.687821000 +0000
@@ -1357,6 +1357,21 @@
 
 /* Get file data from header */
 /* Requires totalfc to be set and label err: to goto on error */
+#if defined(__sgi)
+#define _hgfi(_h, _tag, _td, _flags, _data) \
+    if (headerGet((_h), (_tag), (_td), (_flags))) { \
+	if (rpmtdCount(_td) != totalfc) { \
+	    rpmlog(RPMLOG_ERR, _("Wrong number of entries for tag %s: %u found but %u expected.\n"), rpmTagGetName(_tag), rpmtdCount(_td), totalfc); \
+	    goto err; \
+	} \
+	if (rpmTagGetTagType(_tag) != RPM_STRING_ARRAY_TYPE && rpmTagGetTagType(_tag) != RPM_I18NSTRING_TYPE && \
+	    (_td)->size < totalfc * sizeof(*(_data))) {		\
+	    rpmlog(RPMLOG_ERR, _("Malformed data for tag %s: %u bytes found but %u expected.\n"), rpmTagGetName(_tag), (_td)->size, totalfc * sizeof(*(_data))); \
+	    goto err;				\
+	} \
+	_data = ((_td)->data); \
+    }
+#else
 #define _hgfi(_h, _tag, _td, _flags, _data) \
     if (headerGet((_h), (_tag), (_td), (_flags))) { \
 	if (rpmtdCount(_td) != totalfc) { \
@@ -1370,6 +1385,7 @@
 	} \
 	_data = ((_td)->data); \
     }
+#endif
 /* Get file data from header without checking number of entries */
 #define _hgfinc(_h, _tag, _td, _flags, _data) \
     if (headerGet((_h), (_tag), (_td), (_flags))) {\
diff -N -r -u -w rpm-4.15.0/lib/rpmrc.c rpm-4.15.0-patched/lib/rpmrc.c
--- rpm-4.15.0/lib/rpmrc.c	2019-06-26 15:17:31.416985000 +0000
+++ rpm-4.15.0-patched/lib/rpmrc.c	2019-10-25 17:28:11.694367400 +0000
@@ -1041,6 +1041,32 @@
 	    strcpy(un.machine, __power_pc() ? "ppc" : "rs6000");
 	    sprintf(un.sysname,"aix%s.%s", un.version, un.release);
 	}
+	else if (rstreq(un.sysname, "IRIX64")) {
+#		if __MIPS_SIM == _MIPS_SIM_ABI64
+		    /* 64-bit */
+#			if !defined(__mips_isa_rev) || __mips_isa_rev < 6
+			    /* r1-r5 */
+			    strcpy(un.machine, "mips64");
+#error "mips64"
+#			else
+			    /* r6 */
+			    strcpy(un.machine, "mips64r6");
+#error "mips64r6"
+#			endif
+#		else
+		    /* 32-bit */
+#			if !defined(__mips_isa_rev) || __mips_isa_rev < 6
+			    /* r1-r5 */
+			    strcpy(un.machine, "mips");
+#warning "mips"
+#			else
+			    /* r6 */
+			    strcpy(un.machine, "mipsr6");
+#error "mipsr6"
+#			endif
+#		endif
+	    sprintf(un.sysname, "irix");
+	}
 	else if (rstreq(un.sysname, "Darwin")) { 
 #if defined(__ppc__)
 	    strcpy(un.machine, "ppc");
@@ -1048,7 +1074,7 @@
 	    strcpy(un.machine, "i386");
 #elif defined(__x86_64__)
 	    strcpy(un.machine, "x86_64");
-#else
+#elif !defined(__sgi)
 	    #warning "No architecture defined! Automatic detection may not work!"
 #endif 
 	}
@@ -1075,6 +1101,8 @@
 	for (chptr = un.machine; *chptr != '\0'; chptr++)
 	    if (*chptr == '/') *chptr = '-';
 
+#if !defined(__sgi)
+
 #	if defined(__MIPSEL__) || defined(__MIPSEL) || defined(_MIPSEL)
 	    /* little endian */
 #		if defined(__mips64)
@@ -1119,6 +1147,8 @@
 #		endif
 #	endif
 
+#endif
+
 #if defined(__linux__)
 	/* in linux, lets rename parisc to hppa */
 	if (rstreq(un.machine, "parisc"))
diff -N -r -u -w rpm-4.15.0/lib/rpmscript.c rpm-4.15.0-patched/lib/rpmscript.c
--- rpm-4.15.0/lib/rpmscript.c	2019-06-26 15:17:31.416985000 +0000
+++ rpm-4.15.0-patched/lib/rpmscript.c	2019-10-25 17:28:11.697762360 +0000
@@ -370,8 +370,13 @@
 	   sname, (unsigned)pid, (unsigned)reaped, status);
 
     if (reaped < 0) {
+#if defined(__sgi)
+	rpmlog(lvl, _("%s scriptlet failed, waitpid(%ld) rc %ld: %s\n"),
+		 sname, pid, reaped, strerror(errno));
+#else
 	rpmlog(lvl, _("%s scriptlet failed, waitpid(%d) rc %d: %s\n"),
 		 sname, pid, reaped, strerror(errno));
+#endif
     } else if (!WIFEXITED(status) || WEXITSTATUS(status)) {
       	if (WIFSIGNALED(status)) {
 	    rpmlog(lvl, _("%s scriptlet failed, signal %d\n"),
diff -N -r -u -w rpm-4.15.0/lib/transaction.c rpm-4.15.0-patched/lib/transaction.c
--- rpm-4.15.0/lib/transaction.c	2019-09-09 08:56:53.270788000 +0000
+++ rpm-4.15.0-patched/lib/transaction.c	2019-10-25 17:28:11.704155880 +0000
@@ -1311,7 +1311,11 @@
 
     /* The ordering doesn't matter here */
     /* XXX Only added packages need be checked. */
+#if defined(__sgi)
+    rpmlog(RPMLOG_DEBUG, "sanity checking %llu elements\n", npkgs);
+#else
     rpmlog(RPMLOG_DEBUG, "sanity checking %lu elements\n", npkgs);
+#endif
     pi = rpmtsiInit(ts);
     while ((p = rpmtsiNext(pi, TR_ADDED)) != NULL) {
 
diff -N -r -u -w rpm-4.15.0/luaext/lrexlib.c rpm-4.15.0-patched/luaext/lrexlib.c
--- rpm-4.15.0/luaext/lrexlib.c	2019-06-26 15:17:31.419985000 +0000
+++ rpm-4.15.0-patched/luaext/lrexlib.c	2019-10-25 17:28:11.706730360 +0000
@@ -7,6 +7,10 @@
 #include <stdlib.h>
 #include <string.h>
 
+#if defined(__sgi)
+#include <regex.h>
+#endif
+
 #include "lua.h"
 #include "lauxlib.h"
 #include "lrexlib.h"
@@ -31,7 +35,7 @@
   int res;
   regex_t *pr = (regex_t *)lua_newuserdata(L, sizeof(regex_t));
   pattern = luaL_checklstring(L, 1, &l);
-#ifdef REG_BASIC
+#if defined(REG_BASIC) && !defined(__sgi)
   pr->re_endp = pattern + lua_strlen(L, 1);
   res = regcomp(pr, pattern, REG_EXTENDED | REG_PEND);
 #else
diff -N -r -u -w rpm-4.15.0/macros.in rpm-4.15.0-patched/macros.in
--- rpm-4.15.0/macros.in	2019-09-09 08:56:53.273788000 +0000
+++ rpm-4.15.0-patched/macros.in	2019-10-25 17:28:11.712215560 +0000
@@ -147,7 +147,7 @@
 %_builddir		%{_topdir}/BUILD
 
 #	The interpreter used for build scriptlets.
-%_buildshell		/bin/sh
+%_buildshell		%{_bindir}/sh
 
 #	The path to the bzip2 executable (legacy, use %{__bzip2} instead).
 %_bzip2bin		%{__bzip2}
diff -N -r -u -w rpm-4.15.0/misc/fts.c rpm-4.15.0-patched/misc/fts.c
--- rpm-4.15.0/misc/fts.c	2019-06-26 15:17:31.420985000 +0000
+++ rpm-4.15.0-patched/misc/fts.c	2019-10-25 17:28:11.716813000 +0000
@@ -51,6 +51,12 @@
 #else
 
 /* Conditionals for working around non-GNU environments */
+#if defined(__sgi)
+#include <sys/types.h>
+#define dirfd(dirp)	((dirp)->dd_fd)
+#   define        _INCLUDE_POSIX_SOURCE
+#   define __errno_location() 	(&errno)
+#endif
 #if defined(hpux)
 #   define        _INCLUDE_POSIX_SOURCE
 #   define __errno_location() 	(&errno)
diff -N -r -u -w rpm-4.15.0/python/header-py.c rpm-4.15.0-patched/python/header-py.c
--- rpm-4.15.0/python/header-py.c	2019-09-26 10:44:40.739644000 +0000
+++ rpm-4.15.0-patched/python/header-py.c	2019-10-25 17:28:11.720690040 +0000
@@ -1,3 +1,10 @@
+#if defined(__sgi)
+#include <unistd.h>
+#undef _ABIAPI
+#include <sys/time.h>
+#define _ABIAPI 1
+#endif
+
 #include "rpmsystem-py.h"
 
 #include <rpm/rpmlib.h>		/* rpmvercmp */
diff -N -r -u -w rpm-4.15.0/python/rpmarchive-py.c rpm-4.15.0-patched/python/rpmarchive-py.c
--- rpm-4.15.0/python/rpmarchive-py.c	2019-06-26 15:17:31.443985000 +0000
+++ rpm-4.15.0-patched/python/rpmarchive-py.c	2019-10-25 17:28:11.723031160 +0000
@@ -1,3 +1,10 @@
+#if defined(__sgi)
+#include <unistd.h>
+#undef _ABIAPI
+#include <sys/time.h>
+#define _ABIAPI 1
+#endif
+
 #include "rpmsystem-py.h"
 
 #include <rpm/rpmtypes.h>
diff -N -r -u -w rpm-4.15.0/python/rpmds-py.c rpm-4.15.0-patched/python/rpmds-py.c
--- rpm-4.15.0/python/rpmds-py.c	2019-09-26 10:44:40.739644000 +0000
+++ rpm-4.15.0-patched/python/rpmds-py.c	2019-10-25 17:28:11.725685960 +0000
@@ -1,3 +1,10 @@
+#if defined(__sgi)
+#include <unistd.h>
+#undef _ABIAPI
+#include <sys/time.h>
+#define _ABIAPI 1
+#endif
+
 #include "rpmsystem-py.h"
 
 #include <rpm/rpmtypes.h>
diff -N -r -u -w rpm-4.15.0/python/rpmfd-py.c rpm-4.15.0-patched/python/rpmfd-py.c
--- rpm-4.15.0/python/rpmfd-py.c	2019-09-26 10:44:40.739644000 +0000
+++ rpm-4.15.0-patched/python/rpmfd-py.c	2019-10-25 17:28:11.728195240 +0000
@@ -1,3 +1,9 @@
+#if defined(__sgi)
+#include <unistd.h>
+#undef _ABIAPI
+#include <sys/time.h>
+#define _ABIAPI 1
+#endif
 
 #include "rpmsystem-py.h"
 #include <rpm/rpmstring.h>
diff -N -r -u -w rpm-4.15.0/python/rpmfi-py.c rpm-4.15.0-patched/python/rpmfi-py.c
--- rpm-4.15.0/python/rpmfi-py.c	2019-09-26 10:44:40.739644000 +0000
+++ rpm-4.15.0-patched/python/rpmfi-py.c	2019-10-25 17:28:11.730939000 +0000
@@ -1,3 +1,10 @@
+#if defined(__sgi)
+#include <unistd.h>
+#undef _ABIAPI
+#include <sys/time.h>
+#define _ABIAPI 1
+#endif
+
 #include "rpmsystem-py.h"
 
 #include <rpm/rpmtypes.h>
diff -N -r -u -w rpm-4.15.0/python/rpmfiles-py.c rpm-4.15.0-patched/python/rpmfiles-py.c
--- rpm-4.15.0/python/rpmfiles-py.c	2019-09-26 10:44:40.739644000 +0000
+++ rpm-4.15.0-patched/python/rpmfiles-py.c	2019-10-25 17:28:11.734228120 +0000
@@ -1,3 +1,10 @@
+#if defined(__sgi)
+#include <unistd.h>
+#undef _ABIAPI
+#include <sys/time.h>
+#define _ABIAPI 1
+#endif
+
 #include "rpmsystem-py.h"
 
 #include <rpm/rpmtypes.h>
diff -N -r -u -w rpm-4.15.0/python/rpmii-py.c rpm-4.15.0-patched/python/rpmii-py.c
--- rpm-4.15.0/python/rpmii-py.c	2019-09-26 10:44:40.739644000 +0000
+++ rpm-4.15.0-patched/python/rpmii-py.c	2019-10-25 17:28:11.736292360 +0000
@@ -1,3 +1,10 @@
+#if defined(__sgi)
+#include <unistd.h>
+#undef _ABIAPI
+#include <sys/time.h>
+#define _ABIAPI 1
+#endif
+
 #include "rpmsystem-py.h"
 
 #include <rpm/rpmdb.h>
diff -N -r -u -w rpm-4.15.0/python/rpmkeyring-py.c rpm-4.15.0-patched/python/rpmkeyring-py.c
--- rpm-4.15.0/python/rpmkeyring-py.c	2019-06-26 15:17:31.444985000 +0000
+++ rpm-4.15.0-patched/python/rpmkeyring-py.c	2019-10-25 17:28:11.738429960 +0000
@@ -1,3 +1,10 @@
+#if defined(__sgi)
+#include <unistd.h>
+#undef _ABIAPI
+#include <sys/time.h>
+#define _ABIAPI 1
+#endif
+
 #include "rpmsystem-py.h"
 #include <rpm/rpmkeyring.h>
 #include "rpmkeyring-py.h"
diff -N -r -u -w rpm-4.15.0/python/rpmmacro-py.c rpm-4.15.0-patched/python/rpmmacro-py.c
--- rpm-4.15.0/python/rpmmacro-py.c	2019-06-26 15:17:31.444985000 +0000
+++ rpm-4.15.0-patched/python/rpmmacro-py.c	2019-10-25 17:28:11.740210600 +0000
@@ -1,3 +1,10 @@
+#if defined(__sgi)
+#include <unistd.h>
+#undef _ABIAPI
+#include <sys/time.h>
+#define _ABIAPI 1
+#endif
+
 #include "rpmsystem-py.h"
 
 #include <rpm/rpmmacro.h>
diff -N -r -u -w rpm-4.15.0/python/rpmmi-py.c rpm-4.15.0-patched/python/rpmmi-py.c
--- rpm-4.15.0/python/rpmmi-py.c	2019-09-26 10:44:40.740644000 +0000
+++ rpm-4.15.0-patched/python/rpmmi-py.c	2019-10-25 17:28:11.742566280 +0000
@@ -1,3 +1,10 @@
+#if defined(__sgi)
+#include <unistd.h>
+#undef _ABIAPI
+#include <sys/time.h>
+#define _ABIAPI 1
+#endif
+
 #include "rpmsystem-py.h"
 
 #include <rpm/rpmdb.h>
diff -N -r -u -w rpm-4.15.0/python/rpmmodule.c rpm-4.15.0-patched/python/rpmmodule.c
--- rpm-4.15.0/python/rpmmodule.c	2019-09-26 10:44:40.740644000 +0000
+++ rpm-4.15.0-patched/python/rpmmodule.c	2019-10-25 17:28:11.746268760 +0000
@@ -1,3 +1,10 @@
+#if defined(__sgi)
+#include <unistd.h>
+#undef _ABIAPI
+#include <sys/time.h>
+#define _ABIAPI 1
+#endif
+
 #include "rpmsystem-py.h"
 
 #include <rpm/rpmlib.h>		/* rpmMachineScore, rpmReadConfigFiles */
diff -N -r -u -w rpm-4.15.0/python/rpmps-py.c rpm-4.15.0-patched/python/rpmps-py.c
--- rpm-4.15.0/python/rpmps-py.c	2019-06-26 15:17:31.444985000 +0000
+++ rpm-4.15.0-patched/python/rpmps-py.c	2019-10-25 17:28:11.748364520 +0000
@@ -1,3 +1,10 @@
+#if defined(__sgi)
+#include <unistd.h>
+#undef _ABIAPI
+#include <sys/time.h>
+#define _ABIAPI 1
+#endif
+
 #include "rpmsystem-py.h"
 
 #include "rpmps-py.h"
diff -N -r -u -w rpm-4.15.0/python/rpmstrpool-py.c rpm-4.15.0-patched/python/rpmstrpool-py.c
--- rpm-4.15.0/python/rpmstrpool-py.c	2019-06-26 15:17:31.444985000 +0000
+++ rpm-4.15.0-patched/python/rpmstrpool-py.c	2019-10-25 17:28:11.750346040 +0000
@@ -1,3 +1,10 @@
+#if defined(__sgi)
+#include <unistd.h>
+#undef _ABIAPI
+#include <sys/time.h>
+#define _ABIAPI 1
+#endif
+
 #include "rpmsystem-py.h"
 #include <rpm/rpmstrpool.h>
 #include "rpmstrpool-py.h"
diff -N -r -u -w rpm-4.15.0/python/rpmtd-py.c rpm-4.15.0-patched/python/rpmtd-py.c
--- rpm-4.15.0/python/rpmtd-py.c	2019-06-26 15:17:31.444985000 +0000
+++ rpm-4.15.0-patched/python/rpmtd-py.c	2019-10-25 17:28:11.752298040 +0000
@@ -1,3 +1,10 @@
+#if defined(__sgi)
+#include <unistd.h>
+#undef _ABIAPI
+#include <sys/time.h>
+#define _ABIAPI 1
+#endif
+
 /** \ingroup py_c
  * \file python/rpmtd-py.c
  */
diff -N -r -u -w rpm-4.15.0/python/rpmte-py.c rpm-4.15.0-patched/python/rpmte-py.c
--- rpm-4.15.0/python/rpmte-py.c	2019-06-26 15:17:31.444985000 +0000
+++ rpm-4.15.0-patched/python/rpmte-py.c	2019-10-25 17:28:11.754775320 +0000
@@ -1,3 +1,10 @@
+#if defined(__sgi)
+#include <unistd.h>
+#undef _ABIAPI
+#include <sys/time.h>
+#define _ABIAPI 1
+#endif
+
 #include "rpmsystem-py.h"
 
 #include "header-py.h"	/* XXX tagNumFromPyObject */
diff -N -r -u -w rpm-4.15.0/python/rpmts-py.c rpm-4.15.0-patched/python/rpmts-py.c
--- rpm-4.15.0/python/rpmts-py.c	2019-09-26 10:44:40.740644000 +0000
+++ rpm-4.15.0-patched/python/rpmts-py.c	2019-10-25 17:28:11.759436760 +0000
@@ -1,3 +1,10 @@
+#if defined(__sgi)
+#include <unistd.h>
+#undef _ABIAPI
+#include <sys/time.h>
+#define _ABIAPI 1
+#endif
+
 #include "rpmsystem-py.h"
 
 #include <fcntl.h>
diff -N -r -u -w rpm-4.15.0/python/spec-py.c rpm-4.15.0-patched/python/spec-py.c
--- rpm-4.15.0/python/spec-py.c	2019-06-26 15:17:31.445985000 +0000
+++ rpm-4.15.0-patched/python/spec-py.c	2019-10-25 17:28:11.762316520 +0000
@@ -1,3 +1,10 @@
+#if defined(__sgi)
+#include <unistd.h>
+#undef _ABIAPI
+#include <sys/time.h>
+#define _ABIAPI 1
+#endif
+
 #include "rpmsystem-py.h"
 
 #include "rpmts-py.h"
diff -N -r -u -w rpm-4.15.0/rpmio/macro.c rpm-4.15.0-patched/rpmio/macro.c
--- rpm-4.15.0/rpmio/macro.c	2019-09-09 08:56:53.374788000 +0000
+++ rpm-4.15.0-patched/rpmio/macro.c	2019-10-25 17:28:11.768758120 +0000
@@ -512,7 +512,11 @@
 #endif
     /* Fallback to sysconf() if the above isn't supported or didn't work */
     if (ncpus < 1)
+#if defined(__sgi)
+	ncpus = sysconf(_SC_NPROC_ONLN);
+#else
 	ncpus = sysconf(_SC_NPROCESSORS_ONLN);
+#endif
     /* If all else fails, there's always the one we're running on... */
     if (ncpus < 1)
 	ncpus = 1;
diff -N -r -u -w rpm-4.15.0/rpmio/rpmfileutil.c rpm-4.15.0-patched/rpmio/rpmfileutil.c
--- rpm-4.15.0/rpmio/rpmfileutil.c	2019-06-26 15:17:31.447985000 +0000
+++ rpm-4.15.0-patched/rpmio/rpmfileutil.c	2019-10-25 17:28:11.771605960 +0000
@@ -127,7 +127,11 @@
 	    rc = mkdir(d, mode);
 	    if (rc)
 		goto exit;
+#if defined(__sgi)
+	    rpmlog(RPMLOG_DEBUG, "created directory(s) %s mode 0%lo\n", path, mode);
+#else
 	    rpmlog(RPMLOG_DEBUG, "created directory(s) %s mode 0%o\n", path, mode);
+#endif
 	    if (!(uid == (uid_t) -1 && gid == (gid_t) -1)) {
 		rc = chown(d, uid, gid);
 		if (rc)
diff -N -r -u -w rpm-4.15.0/rpmio/rpmsq.c rpm-4.15.0-patched/rpmio/rpmsq.c
--- rpm-4.15.0/rpmio/rpmsq.c	2019-06-26 15:17:31.450985000 +0000
+++ rpm-4.15.0-patched/rpmio/rpmsq.c	2019-10-25 17:28:11.773831480 +0000
@@ -34,7 +34,11 @@
     } else {
 	int lvl = (signum == SIGPIPE) ? RPMLOG_DEBUG : RPMLOG_WARNING;
 	rpmlog(lvl,
+#if defined(__sgi)
+		_("exiting on signal %d from pid %ld\n"), signum, info->si_pid);
+#else
 		_("exiting on signal %d from pid %d\n"), signum, info->si_pid);
+#endif
     }
     /* exit 128 + signum for compatibility with bash(1) */
     exit(128 + signum);
diff -N -r -u -w rpm-4.15.0/scripts/brp-compress rpm-4.15.0-patched/scripts/brp-compress
--- rpm-4.15.0/scripts/brp-compress	2019-09-09 08:56:53.375788000 +0000
+++ rpm-4.15.0-patched/scripts/brp-compress	2019-10-25 17:28:11.775584280 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env sh
 
 # If using normal root, avoid changing anything.
 if [ -z "$RPM_BUILD_ROOT" ] || [ "$RPM_BUILD_ROOT" = "/" ]; then
diff -N -r -u -w rpm-4.15.0/scripts/brp-java-gcjcompile rpm-4.15.0-patched/scripts/brp-java-gcjcompile
--- rpm-4.15.0/scripts/brp-java-gcjcompile	2019-09-09 08:56:53.375788000 +0000
+++ rpm-4.15.0-patched/scripts/brp-java-gcjcompile	2019-10-25 17:28:11.777446680 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env sh
 
 # If using normal root, avoid changing anything.
 [ -z "$RPM_BUILD_ROOT" ] || [ "$RPM_BUILD_ROOT" = "/" ] && exit 0
diff -N -r -u -w rpm-4.15.0/scripts/brp-python-bytecompile rpm-4.15.0-patched/scripts/brp-python-bytecompile
--- rpm-4.15.0/scripts/brp-python-bytecompile	2019-09-09 08:56:53.375788000 +0000
+++ rpm-4.15.0-patched/scripts/brp-python-bytecompile	2019-10-25 19:06:29.897539320 +0000
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
 errors_terminate=$2
 extra=$3
 
diff -N -r -u -w rpm-4.15.0/scripts/brp-python-hardlink rpm-4.15.0-patched/scripts/brp-python-hardlink
--- rpm-4.15.0/scripts/brp-python-hardlink	2019-09-09 08:56:53.375788000 +0000
+++ rpm-4.15.0-patched/scripts/brp-python-hardlink	2019-10-25 17:28:11.779156840 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env sh
 
 # If using normal root, avoid changing anything.
 if [ -z "$RPM_BUILD_ROOT" ] || [ "$RPM_BUILD_ROOT" = "/" ]; then
diff -N -r -u -w rpm-4.15.0/scripts/brp-strip rpm-4.15.0-patched/scripts/brp-strip
--- rpm-4.15.0/scripts/brp-strip	2019-09-09 08:56:53.375788000 +0000
+++ rpm-4.15.0-patched/scripts/brp-strip	2019-10-25 17:28:11.780815000 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env sh
 # If using normal root, avoid changing anything.
 if [ -z "$RPM_BUILD_ROOT" ] || [ "$RPM_BUILD_ROOT" = "/" ]; then
 	exit 0
diff -N -r -u -w rpm-4.15.0/scripts/brp-strip-comment-note rpm-4.15.0-patched/scripts/brp-strip-comment-note
--- rpm-4.15.0/scripts/brp-strip-comment-note	2019-09-09 08:56:53.375788000 +0000
+++ rpm-4.15.0-patched/scripts/brp-strip-comment-note	2019-10-25 17:28:11.782477640 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env sh
 # If using normal root, avoid changing anything.
 if [ -z "$RPM_BUILD_ROOT" ] || [ "$RPM_BUILD_ROOT" = "/" ]; then
 	exit 0
diff -N -r -u -w rpm-4.15.0/scripts/brp-strip-shared rpm-4.15.0-patched/scripts/brp-strip-shared
--- rpm-4.15.0/scripts/brp-strip-shared	2019-09-09 08:56:53.376788000 +0000
+++ rpm-4.15.0-patched/scripts/brp-strip-shared	2019-10-25 17:28:11.784170040 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env sh
 # Conectiva brp - strip shared libraries. Based on Red Hat's brp-strip.
 # Thu Apr 20 - Guilherme Manika <gwm@conectiva.com.br>
 #              Created file
diff -N -r -u -w rpm-4.15.0/scripts/brp-strip-static-archive rpm-4.15.0-patched/scripts/brp-strip-static-archive
--- rpm-4.15.0/scripts/brp-strip-static-archive	2019-09-09 08:56:53.376788000 +0000
+++ rpm-4.15.0-patched/scripts/brp-strip-static-archive	2019-10-25 17:28:11.785855640 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env sh
 
 if [ -z "$RPM_BUILD_ROOT" ] || [ "$RPM_BUILD_ROOT" = "/" ]; then
 	exit 0
diff -N -r -u -w rpm-4.15.0/scripts/check-buildroot rpm-4.15.0-patched/scripts/check-buildroot
--- rpm-4.15.0/scripts/check-buildroot	2019-09-09 08:56:53.376788000 +0000
+++ rpm-4.15.0-patched/scripts/check-buildroot	2019-10-25 17:28:11.787622280 +0000
@@ -1,4 +1,4 @@
-#! /bin/sh
+#! /usr/bin/env sh
 
 # Copyright (C) 2004 Enrico Scholz <enrico.scholz@informatik.tu-chemnitz.de>
 #  
diff -N -r -u -w rpm-4.15.0/scripts/check-files rpm-4.15.0-patched/scripts/check-files
--- rpm-4.15.0/scripts/check-files	2019-06-26 15:17:31.452985000 +0000
+++ rpm-4.15.0-patched/scripts/check-files	2019-10-25 17:28:11.789316200 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env sh
 #
 # Gets file list on standard input and RPM_BUILD_ROOT as first parameter
 # and searches for omitted files (not counting directories).
diff -N -r -u -w rpm-4.15.0/scripts/check-prereqs rpm-4.15.0-patched/scripts/check-prereqs
--- rpm-4.15.0/scripts/check-prereqs	2019-06-26 15:17:31.452985000 +0000
+++ rpm-4.15.0-patched/scripts/check-prereqs	2019-10-25 19:05:09.697529400 +0000
@@ -1,6 +1,6 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
-bashit="/bin/bash --rpm-requires"
+bashit="/usr/bin/env bash --rpm-requires"
 
 # Make sure that this bash has the rpm-requires hack
 $bashit < /dev/null 2>&1 > /dev/null || exit $?
diff -N -r -u -w rpm-4.15.0/scripts/check-rpaths rpm-4.15.0-patched/scripts/check-rpaths
--- rpm-4.15.0/scripts/check-rpaths	2019-09-09 08:56:53.376788000 +0000
+++ rpm-4.15.0-patched/scripts/check-rpaths	2019-10-25 17:28:11.791090360 +0000
@@ -1,4 +1,4 @@
-#! /bin/sh
+#! /usr/bin/env sh
 
 # Copyright (C) 2004 Enrico Scholz <enrico.scholz@informatik.tu-chemnitz.de>
 #  
diff -N -r -u -w rpm-4.15.0/scripts/check-rpaths-worker rpm-4.15.0-patched/scripts/check-rpaths-worker
--- rpm-4.15.0/scripts/check-rpaths-worker	2019-06-26 15:17:31.452985000 +0000
+++ rpm-4.15.0-patched/scripts/check-rpaths-worker	2019-10-25 19:07:20.170226920 +0000
@@ -1,4 +1,4 @@
-#! /bin/bash
+#! /usr/bin/env bash
 
 # Copyright (C) 2004 Enrico Scholz <enrico.scholz@informatik.tu-chemnitz.de>
 #  
diff -N -r -u -w rpm-4.15.0/scripts/debuginfo.prov rpm-4.15.0-patched/scripts/debuginfo.prov
--- rpm-4.15.0/scripts/debuginfo.prov	2019-06-26 15:17:31.452985000 +0000
+++ rpm-4.15.0-patched/scripts/debuginfo.prov	2019-10-25 17:28:11.792745400 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env sh
 
 while read instfile; do
   case "$instfile" in
diff -N -r -u -w rpm-4.15.0/scripts/desktop-file.prov rpm-4.15.0-patched/scripts/desktop-file.prov
--- rpm-4.15.0/scripts/desktop-file.prov	2019-06-26 15:17:31.452985000 +0000
+++ rpm-4.15.0-patched/scripts/desktop-file.prov	2019-10-25 17:28:11.794431080 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env sh
 #
 # Transform desktop mimetype info into RPM mimehandler(type) provides
 #
diff -N -r -u -w rpm-4.15.0/scripts/find-debuginfo.sh rpm-4.15.0-patched/scripts/find-debuginfo.sh
--- rpm-4.15.0/scripts/find-debuginfo.sh	2019-09-09 08:56:53.377788000 +0000
+++ rpm-4.15.0-patched/scripts/find-debuginfo.sh	2019-10-25 19:06:53.890457000 +0000
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
 #find-debuginfo.sh - automagically generate debug info and file list
 #for inclusion in an rpm spec file.
 #
diff -N -r -u -w rpm-4.15.0/scripts/find-lang.sh rpm-4.15.0-patched/scripts/find-lang.sh
--- rpm-4.15.0/scripts/find-lang.sh	2019-09-09 08:56:53.377788000 +0000
+++ rpm-4.15.0-patched/scripts/find-lang.sh	2019-10-25 19:04:37.246990880 +0000
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
 #findlang - automagically generate list of language specific files
 #for inclusion in an rpm spec file.
 #This does assume that the *.mo files are under .../locale/...
diff -N -r -u -w rpm-4.15.0/scripts/find-provides rpm-4.15.0-patched/scripts/find-provides
--- rpm-4.15.0/scripts/find-provides	2019-06-26 15:17:31.453985000 +0000
+++ rpm-4.15.0-patched/scripts/find-provides	2019-10-25 17:28:11.796073320 +0000
@@ -1,3 +1,3 @@
-#!/bin/sh
+#!/usr/bin/env sh
 
 /usr/lib/rpm/rpmdeps --define="_use_internal_dependency_generator 1" --provides
diff -N -r -u -w rpm-4.15.0/scripts/find-provides.php rpm-4.15.0-patched/scripts/find-provides.php
--- rpm-4.15.0/scripts/find-provides.php	2019-06-26 15:17:31.453985000 +0000
+++ rpm-4.15.0-patched/scripts/find-provides.php	2019-10-25 17:28:11.797699000 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env sh
 if [ $# -lt 1 ]; then
 	echo "You have to specify input file"
 	exit 1
diff -N -r -u -w rpm-4.15.0/scripts/find-requires rpm-4.15.0-patched/scripts/find-requires
--- rpm-4.15.0/scripts/find-requires	2019-06-26 15:17:31.453985000 +0000
+++ rpm-4.15.0-patched/scripts/find-requires	2019-10-25 17:28:11.799346600 +0000
@@ -1,3 +1,3 @@
-#!/bin/sh
+#!/usr/bin/env sh
 
 /usr/lib/rpm/rpmdeps --define="_use_internal_dependency_generator 1" --requires
diff -N -r -u -w rpm-4.15.0/scripts/find-requires.php rpm-4.15.0-patched/scripts/find-requires.php
--- rpm-4.15.0/scripts/find-requires.php	2019-06-26 15:17:31.453985000 +0000
+++ rpm-4.15.0-patched/scripts/find-requires.php	2019-10-25 17:28:11.801187960 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env sh
 #####################################################################
 #                                                                   #
 # Check system dependences between php-pear modules                 #
diff -N -r -u -w rpm-4.15.0/scripts/fontconfig.prov rpm-4.15.0-patched/scripts/fontconfig.prov
--- rpm-4.15.0/scripts/fontconfig.prov	2019-06-26 15:17:31.453985000 +0000
+++ rpm-4.15.0-patched/scripts/fontconfig.prov	2019-10-25 19:07:07.758089480 +0000
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
 #
 # Script to install in:
 # /usr/lib/rpm/redhat/find-provides.d
diff -N -r -u -w rpm-4.15.0/scripts/gendiff rpm-4.15.0-patched/scripts/gendiff
--- rpm-4.15.0/scripts/gendiff	2019-09-09 08:56:53.377788000 +0000
+++ rpm-4.15.0-patched/scripts/gendiff	2019-10-25 17:28:11.802871800 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env sh
 
 [ -z "$1" ] || [ -z "$2" ] && {
 # usage
diff -N -r -u -w rpm-4.15.0/scripts/libtooldeps.sh rpm-4.15.0-patched/scripts/libtooldeps.sh
--- rpm-4.15.0/scripts/libtooldeps.sh	2019-06-26 15:17:31.453985000 +0000
+++ rpm-4.15.0-patched/scripts/libtooldeps.sh	2019-10-25 17:28:11.804549960 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env sh
 
 [ $# -ge 2 ] || {
     cat > /dev/null
diff -N -r -u -w rpm-4.15.0/scripts/metainfo.prov rpm-4.15.0-patched/scripts/metainfo.prov
--- rpm-4.15.0/scripts/metainfo.prov	2019-06-26 15:17:31.453985000 +0000
+++ rpm-4.15.0-patched/scripts/metainfo.prov	2019-10-25 17:28:11.806307720 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env sh
 #
 # Transform appdata/metainfo xml file into RPM metainfo(filename) provides
 #
diff -N -r -u -w rpm-4.15.0/scripts/ocaml-find-provides.sh rpm-4.15.0-patched/scripts/ocaml-find-provides.sh
--- rpm-4.15.0/scripts/ocaml-find-provides.sh	2019-06-26 15:17:31.453985000 +0000
+++ rpm-4.15.0-patched/scripts/ocaml-find-provides.sh	2019-10-25 17:28:11.808150280 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh -
+#!/usr/bin/env sh -
 # OCaml-specific "find-provides" for RPM.
 # By Richard W.M. Jones <rjones@redhat.com>
 # $Id: ocaml-find-provides.sh,v 1.2 2007/09/06 11:49:59 rjones Exp $
diff -N -r -u -w rpm-4.15.0/scripts/ocaml-find-requires.sh rpm-4.15.0-patched/scripts/ocaml-find-requires.sh
--- rpm-4.15.0/scripts/ocaml-find-requires.sh	2019-06-26 15:17:31.453985000 +0000
+++ rpm-4.15.0-patched/scripts/ocaml-find-requires.sh	2019-10-25 17:28:11.809989160 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh -
+#!/usr/bin/env sh -
 # OCaml-specific "find-requires" for RPM.
 # By Richard W.M. Jones <rjones@redhat.com>
 # $Id: ocaml-find-requires.sh,v 1.5 2009/10/04 22:34:51 rjones Exp $
diff -N -r -u -w rpm-4.15.0/scripts/perl.prov rpm-4.15.0-patched/scripts/perl.prov
--- rpm-4.15.0/scripts/perl.prov	2019-06-26 15:17:31.453985000 +0000
+++ rpm-4.15.0-patched/scripts/perl.prov	2019-10-25 17:28:11.812117240 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl
 
 # RPM (and it's source code) is covered under two separate licenses.
 
diff -N -r -u -w rpm-4.15.0/scripts/perl.req rpm-4.15.0-patched/scripts/perl.req
--- rpm-4.15.0/scripts/perl.req	2019-06-26 15:17:31.453985000 +0000
+++ rpm-4.15.0-patched/scripts/perl.req	2019-10-25 17:28:11.814554040 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl
 
 # RPM (and its source code) is covered under two separate licenses.
 
diff -N -r -u -w rpm-4.15.0/scripts/pkgconfigdeps.sh rpm-4.15.0-patched/scripts/pkgconfigdeps.sh
--- rpm-4.15.0/scripts/pkgconfigdeps.sh	2019-06-26 15:17:31.453985000 +0000
+++ rpm-4.15.0-patched/scripts/pkgconfigdeps.sh	2019-10-25 19:04:48.456044560 +0000
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
 pkgconfig=/usr/bin/pkg-config
 test -x $pkgconfig || {
diff -N -r -u -w rpm-4.15.0/scripts/pythondeps.sh rpm-4.15.0-patched/scripts/pythondeps.sh
--- rpm-4.15.0/scripts/pythondeps.sh	2019-06-26 15:17:31.453985000 +0000
+++ rpm-4.15.0-patched/scripts/pythondeps.sh	2019-10-25 19:06:16.185045240 +0000
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
 [ $# -ge 1 ] || {
     cat > /dev/null
diff -N -r -u -w rpm-4.15.0/scripts/rpm.daily rpm-4.15.0-patched/scripts/rpm.daily
--- rpm-4.15.0/scripts/rpm.daily	2019-06-26 15:17:31.454985000 +0000
+++ rpm-4.15.0-patched/scripts/rpm.daily	2019-10-25 17:28:11.816202680 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env sh
 
 tmpfile=`/bin/mktemp /var/log/rpmpkgs.XXXXXXXXX` || exit 1
 /bin/rpm -qa --qf '%{name}-%{version}-%{release}.%{arch}.rpm\n' 2>&1 \
diff -N -r -u -w rpm-4.15.0/scripts/rpm2cpio.sh rpm-4.15.0-patched/scripts/rpm2cpio.sh
--- rpm-4.15.0/scripts/rpm2cpio.sh	2019-09-09 08:56:53.377788000 +0000
+++ rpm-4.15.0-patched/scripts/rpm2cpio.sh	2019-10-25 17:28:11.817885080 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh -efu
+#!/usr/bin/env sh -efu
 
 fatal() {
 	echo "$*" >&2
diff -N -r -u -w rpm-4.15.0/scripts/rpmdb_loadcvt rpm-4.15.0-patched/scripts/rpmdb_loadcvt
--- rpm-4.15.0/scripts/rpmdb_loadcvt	2019-09-09 08:56:53.377788000 +0000
+++ rpm-4.15.0-patched/scripts/rpmdb_loadcvt	2019-10-25 19:06:40.241070840 +0000
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
 ac=$#
 cmd=`basename $0`
diff -N -r -u -w rpm-4.15.0/scripts/script.req rpm-4.15.0-patched/scripts/script.req
--- rpm-4.15.0/scripts/script.req	2019-06-26 15:17:31.454985000 +0000
+++ rpm-4.15.0-patched/scripts/script.req	2019-10-25 17:28:11.819548840 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env sh
 
 # TODO: handle "#!/usr/bin/env foo" somehow
 while read filename; do
diff -N -r -u -w rpm-4.15.0/scripts/tgpg rpm-4.15.0-patched/scripts/tgpg
--- rpm-4.15.0/scripts/tgpg	2019-09-09 08:56:53.377788000 +0000
+++ rpm-4.15.0-patched/scripts/tgpg	2019-10-25 17:28:11.821244760 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env sh
 
 
 for pkg in $*
diff -N -r -u -w rpm-4.15.0/scripts/vpkg-provides.sh rpm-4.15.0-patched/scripts/vpkg-provides.sh
--- rpm-4.15.0/scripts/vpkg-provides.sh	2019-06-26 15:17:31.454985000 +0000
+++ rpm-4.15.0-patched/scripts/vpkg-provides.sh	2019-10-25 17:28:11.823688120 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env sh
 
 #
 # Original Author: Tim Mooney (mooney@plains.NoDak.edu)
diff -N -r -u -w rpm-4.15.0/system.h rpm-4.15.0-patched/system.h
--- rpm-4.15.0/system.h	2019-06-26 15:17:31.455985000 +0000
+++ rpm-4.15.0-patched/system.h	2019-10-25 17:28:11.825611080 +0000
@@ -102,6 +102,9 @@
 # include <stdlib.h> /* Make sure this header is included */
 # define xsetprogname(pn) setprogname(pn)
 # define xgetprogname(pn) getprogname(pn)
+#elif defined(__sgi)
+# define xsetprogname(pn)
+# define xgetprogname(pn) getprogname(pn)
 #elif defined(HAVE___PROGNAME) /* glibc and others */
 # define xsetprogname(pn)
   extern const char *__progname;
diff -N -r -u -w rpm-4.15.0/tools/debugedit.c rpm-4.15.0-patched/tools/debugedit.c
--- rpm-4.15.0/tools/debugedit.c	2019-06-26 15:17:31.462985000 +0000
+++ rpm-4.15.0-patched/tools/debugedit.c	2019-10-25 17:28:11.835329800 +0000
@@ -19,6 +19,11 @@
 
 #include "system.h"
 
+#if defined(__sgi)
+#include <stdlib.h>
+#include <alloca.h>
+#endif
+
 /* Needed for libelf */
 #define _FILE_OFFSET_BITS 64
 
@@ -1106,6 +1111,14 @@
   strings->strent_root = NULL;
 }
 
+#if defined(__sgi)
+void tdestroy (void *root, void(*free_node)(void*nodep))
+{
+  fprintf(stdout,"EMTPY TDESTROY ISN'T FREEING STUFF, MR HAMS\n");
+  fflush(stdout);
+}
+#endif
+
 /* Noop for tdestroy. */
 static void free_node (void *p __attribute__((__unused__))) { }
 
diff -N -r -u -w rpm-4.15.0/tools/sepdebugcrcfix.c rpm-4.15.0-patched/tools/sepdebugcrcfix.c
--- rpm-4.15.0/tools/sepdebugcrcfix.c	2019-06-26 15:17:31.463985000 +0000
+++ rpm-4.15.0-patched/tools/sepdebugcrcfix.c	2019-10-25 17:28:11.838272520 +0000
@@ -154,8 +154,13 @@
       void *map = NULL;
       if (buf == NULL)
 	{
+#if defined(__sgi)
+	  map = mmap (NULL, maplen, PROT_READ, MAP_PRIVATE,
+		      fd, offset);
+#else
 	  map = mmap (NULL, maplen, PROT_READ, MAP_PRIVATE | MAP_POPULATE,
 		      fd, offset);
+#endif
 	  if (map == MAP_FAILED)
 	    {
 	      error (0, errno, _("cannot map 0x%llx bytes at offset 0x%llx "
