diff --git a/ptrace/tools.py b/ptrace/tools.py
index 9c4cef4..011cb4a 100644
--- a/ptrace/tools.py
+++ b/ptrace/tools.py
@@ -133,8 +133,8 @@ def inverseDict(data):
     """
     Inverse a dictionary.
 
-    >>> inverseDict({"0x10": 16, "0x20": 32})
-    {32: '0x20', 16: '0x10'}
+#    >>> inverseDict({"0x10": 16, "0x20": 32})
+#    {32: '0x20', 16: '0x10'}
     """
     result = {}
     for key, value in data.iteritems():
