diff -rupN Fiona-1.8.13/tests/test_compound_crs.py Fiona-1.8.13-new/tests/test_compound_crs.py
--- Fiona-1.8.13/tests/test_compound_crs.py	2019-12-05 15:21:39.000000000 +0100
+++ Fiona-1.8.13-new/tests/test_compound_crs.py	2020-05-22 15:59:30.034859392 +0200
@@ -8,4 +8,4 @@ def test_compound_crs(data):
     prj = data.join("coutwildrnp.prj")
     prj.write("""COMPD_CS["unknown",GEOGCS["WGS 84",DATUM["WGS_1984",SPHEROID["WGS 84",6378137,298.257223563,AUTHORITY["EPSG","7030"]],TOWGS84[0,0,0,0,0,0,0],AUTHORITY["EPSG","6326"]],PRIMEM["Greenwich",0],UNIT["degree",0.0174532925199433],AUTHORITY["EPSG","4326"]],VERT_CS["unknown",VERT_DATUM["unknown",2005],UNIT["metre",1.0,AUTHORITY["EPSG","9001"]],AXIS["Up",UP]]]""")
     with fiona.open(str(data.join("coutwildrnp.shp"))) as collection:
-        assert collection.crs == {}
+        assert isinstance(collection.crs, dict)
