Index: argh-0.26.1/test/test_integration.py
===================================================================
--- argh-0.26.1.orig/test/test_integration.py
+++ argh-0.26.1/test/test_integration.py
@@ -772,4 +772,4 @@ def test_unknown_args():
     assert run(p, '--bar 1', exit=True) == 'unrecognized arguments: --bar 1'
     assert run(p, '--bar 1', exit=False,
                kwargs={'skip_unknown_args': True}) == \
-           R(out='usage: py.test [-h] [-f FOO]\n\n', err='')
+           R(out='usage: setup.py [-h] [-f FOO]\n\n', err='')
