diff -ur glu-9.0.1.orig/src/libnurbs/interface/glrenderer.h glu-9.0.1/src/libnurbs/interface/glrenderer.h
--- glu-9.0.1.orig/src/libnurbs/interface/glrenderer.h	2021-01-01 15:28:38.650175200 +0000
+++ glu-9.0.1/src/libnurbs/interface/glrenderer.h	2021-01-01 15:29:18.463885600 +0000
@@ -42,6 +42,10 @@
 #include "glsurfeval.h"
 #include "glcurveval.h"
 
+#ifndef APIENTRY
+#define APIENTRY
+#endif
+
 extern "C" {
       typedef void (APIENTRY *errorCallbackType)( GLenum );
 }
