diff -u -r modulemd-2.9.3-orig/modulemd/modulemd-compression.c modulemd-2.9.3/modulemd/modulemd-compression.c
--- modulemd-2.9.3-orig/modulemd/modulemd-compression.c	2020-04-08 15:38:27.207601000 +0000
+++ modulemd-2.9.3/modulemd/modulemd-compression.c	2020-08-16 16:31:39.604800720 +0000
@@ -82,7 +82,11 @@
   /* No known suffix? Try using libmagic from file-utils */
   const char *mime_type;
   g_auto (magic_t) magic = NULL;
+#if defined(__sgi)
+  int magic_fd = fcntl (fd, FD_CLOEXEC, 0);
+#else
   int magic_fd = fcntl (fd, F_DUPFD_CLOEXEC, 0);
+#endif
   int err = errno;
   if (magic_fd < 0)
     {
diff -u -r modulemd-2.9.3-orig/modulemd/modulemd-module-index.c modulemd-2.9.3/modulemd/modulemd-module-index.c
--- modulemd-2.9.3-orig/modulemd/modulemd-module-index.c	2020-04-08 15:38:27.212601000 +0000
+++ modulemd-2.9.3/modulemd/modulemd-module-index.c	2020-08-16 16:34:36.903151760 +0000
@@ -528,6 +528,10 @@
 }
 
 
+#if defined(__sgi)
+extern int fileno(FILE *stream);
+#endif
+
 gboolean
 modulemd_module_index_update_from_file (ModulemdModuleIndex *self,
                                         const gchar *yaml_file,
diff -u -r modulemd-2.9.3-orig/modulemd/tests/test-modulemd-compression.c modulemd-2.9.3/modulemd/tests/test-modulemd-compression.c
--- modulemd-2.9.3-orig/modulemd/tests/test-modulemd-compression.c	2020-04-08 15:38:27.232601000 +0000
+++ modulemd-2.9.3/modulemd/tests/test-modulemd-compression.c	2020-08-16 16:38:19.098197360 +0000
@@ -61,6 +61,10 @@
   ModulemdCompressionTypeEnum type;
 };
 
+#if defined(__sgi)
+extern int fileno(FILE *stream);
+#endif
+
 /* == Private Functions == */
 static void
 test_modulemd_detect_compression (void)
