diff -Naur gegl-0.4.26.orig/gegl/gegl-config.c gegl-0.4.26/gegl/gegl-config.c
--- gegl-0.4.26.orig/gegl/gegl-config.c	2020-08-02 13:23:44.906186000 +0000
+++ gegl-0.4.26/gegl/gegl-config.c	2020-12-27 23:08:16.394128685 +0000
@@ -29,7 +29,7 @@
 #include "gegl-types-internal.h"
 #include "gegl-config.h"
 #include "gegl-buffer-config.h"
-
+#include <sys/sysmp.h>
 #include "opencl/gegl-cl.h"
 
 #ifdef HAVE_UNISTD_H
@@ -317,9 +317,12 @@
                               - (uint64_t) laundry_count * page_size
                               + zfs_arc_size;
     }
-#else
-    mem_total = (uint64_t) sysconf (_SC_PHYS_PAGES) * sysconf (_SC_PAGESIZE);
-    mem_available = (uint64_t) sysconf (_SC_AVPHYS_PAGES) * sysconf (_SC_PAGESIZE);
+#elif defined(__sgi)
+struct rminfo meminfo;
+if (sysmp(MP_SAGET, MPSA_RMINFO, &meminfo, sizeof(meminfo)) == 0) {
+  mem_total = meminfo.physmem;
+  mem_available = meminfo.availrmem;
+}
 #endif
 
     default_tile_cache_size = mem_total;
