Index: glog-0.3.5/src/raw_logging.cc
===================================================================
--- glog-0.3.5.orig/src/raw_logging.cc
+++ glog-0.3.5/src/raw_logging.cc
@@ -59,12 +59,8 @@
 # include <unistd.h>
 #endif
 
-#if defined(HAVE_SYSCALL_H) || defined(HAVE_SYS_SYSCALL_H)
-# define safe_write(fd, s, len)  syscall(SYS_write, fd, s, len)
-#else
-  // Not so safe, but what can you do?
+// Not so safe, but what can you do?
 # define safe_write(fd, s, len)  write(fd, s, len)
-#endif
 
 _START_GOOGLE_NAMESPACE_
 
