diff -u -r libjpeg-turbo-2.0.2-orig/md5/md5.h libjpeg-turbo-2.0.2/md5/md5.h
--- libjpeg-turbo-2.0.2-orig/md5/md5.h	2019-02-14 04:03:31.000000000 +0000
+++ libjpeg-turbo-2.0.2/md5/md5.h	2020-10-27 01:42:25.587157240 +0000
@@ -33,6 +33,8 @@
 #include <sys/types.h>
 #ifdef __amigaos4__
 #include <machine/endian.h>
+#elif defined(__sgi)
+#include <sys/endian.h>
 #endif
 
 /*  On machines where "long" is 64 bits, we need to declare
