From 44aeb741aba986fcc6b1bceb85d49131d35e00f5 Mon Sep 17 00:00:00 2001
From: Elliott Sales de Andrade <quantum.analyst@gmail.com>
Date: Sun, 29 Jul 2018 19:34:39 -0400
Subject: [PATCH] Skip test_overlay_nybb on broken platforms.

Signed-off-by: Elliott Sales de Andrade <quantum.analyst@gmail.com>
---
 geopandas/tests/test_overlay.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/geopandas/tests/test_overlay.py b/geopandas/tests/test_overlay.py
index b344fed..b40e9b3 100644
--- a/geopandas/tests/test_overlay.py
+++ b/geopandas/tests/test_overlay.py
@@ -1,4 +1,5 @@
 import os
+import platform
 
 import pandas as pd
 
@@ -100,6 +101,9 @@ def test_overlay(dfs_index, how):
         assert_geodataframe_equal(result, expected, check_column_type=False)
 
 
+@pytest.mark.xfail(
+    platform.machine() in ('aarch64', 'ppc64', 'ppc64le', 's390x'),
+    reason='platform-specific problems')
 @pytest.mark.filterwarnings("ignore:GeoSeries crs mismatch:UserWarning")
 def test_overlay_nybb(how):
     polydf = read_file(geopandas.datasets.get_path("nybb"))
-- 
2.21.1

