diff -Naur exiv2-0.26.orig/CMakeLists.txt exiv2-0.26/CMakeLists.txt
--- exiv2-0.26.orig/CMakeLists.txt	2020-05-25 10:16:11.814214000 +0000
+++ exiv2-0.26/CMakeLists.txt	2020-05-25 10:17:06.800179423 +0000
@@ -215,8 +215,8 @@
 #   )
 # ENDIF()
 
-include(config/generateDoc.cmake REQUIRED)
-generate_documentation("${PROJECT_SOURCE_DIR}/config/Doxyfile")
+#include(config/generateDoc.cmake REQUIRED)
+#generate_documentation("${PROJECT_SOURCE_DIR}/config/Doxyfile")
 
 # That's all Folks!
 ##
