diff -u -r glib-2.62.6-orig/gio/glocalfile.c glib-2.62.6/gio/glocalfile.c
--- glib-2.62.6-orig/gio/glocalfile.c	2020-03-18 13:16:11.000000000 +0000
+++ glib-2.62.6/gio/glocalfile.c	2020-05-31 12:52:34.740537560 +0000
@@ -2734,7 +2734,7 @@
   if (g_cancellable_set_error_if_cancelled (state->cancellable, error))
     return FALSE;
 
-#if defined (AT_FDCWD)
+#if defined (AT_FDCWD) && !defined(__sgi)
   if (fstatat (parent_fd, name->data, &buf, AT_SYMLINK_NOFOLLOW) != 0)
     {
       int errsv = errno;
@@ -2826,7 +2826,7 @@
       if (g_cancellable_set_error_if_cancelled (state->cancellable, error))
         return FALSE;
 
-#ifdef AT_FDCWD
+#if defined(AT_FDCWD) && !defined(__sgi)
 #ifdef HAVE_OPEN_O_DIRECTORY
       dir_fd = openat (parent_fd, name->data, O_RDONLY|O_DIRECTORY);
 #else
@@ -2854,7 +2854,7 @@
   const gchar *name;
   GDir *dir;
 
-#ifdef AT_FDCWD
+#if defined(AT_FDCWD) && !defined(__sgi)
   {
     /* If this fails, we want to preserve the errno from fopendir() */
     DIR *dirp;
diff -u -r glib-2.62.6-orig/gio/gsocket.c glib-2.62.6/gio/gsocket.c
--- glib-2.62.6-orig/gio/gsocket.c	2020-03-18 13:16:11.000000000 +0000
+++ glib-2.62.6/gio/gsocket.c	2020-05-31 12:52:34.749592520 +0000
@@ -4427,7 +4427,7 @@
   const struct msghdr *_prev_msg = (prev_msg); \
   GError **_error = (error); \
  \
-  _msg->msg_flags = 0; \
+  /*_msg->msg_flags = 0;*/                      \
  \
   /* name */ \
   if (_prev_message != NULL && _prev_message->address == _message->address) \
@@ -4568,7 +4568,7 @@
     } \
  \
   /* flags */ \
-  _msg->msg_flags = _message->flags; \
+  /*_msg->msg_flags = _message->flags;*/        \
 } G_STMT_END
 
 static void
@@ -4634,7 +4634,7 @@
   }
 
   /* capture the flags */
-  message->flags = msg->msg_flags;
+  /*message->flags = msg->msg_flags;*/
 }
 #endif
 
@@ -5336,7 +5336,11 @@
     /* do it */
     while (1)
       {
+#if defined(__sgi)
+	result = recvmsg (socket->priv->fd, &msg, 0);
+#else
 	result = recvmsg (socket->priv->fd, &msg, msg.msg_flags);
+#endif
 #ifdef MSG_CMSG_CLOEXEC	
 	if (result < 0 && get_socket_errno () == EINVAL)
 	  {
diff -u -r glib-2.62.6-orig/gio/gtrashportal.c glib-2.62.6/gio/gtrashportal.c
--- glib-2.62.6-orig/gio/gtrashportal.c	2020-03-18 13:16:11.000000000 +0000
+++ glib-2.62.6/gio/gtrashportal.c	2020-05-31 12:52:34.751526040 +0000
@@ -86,10 +86,17 @@
 
   path = g_file_get_path (file);
 
+#if defined(__sgi)
+  fd = g_open (path, O_RDWR | O_CLOEXEC);
+  if (fd == -1 && errno == EISDIR)
+    /* If it is a directory, fall back to O_PATH */
+    fd = g_open (path, O_PATH | O_CLOEXEC | O_RDONLY);
+#else
   fd = g_open (path, O_RDWR | O_CLOEXEC | O_NOFOLLOW);
   if (fd == -1 && errno == EISDIR)
     /* If it is a directory, fall back to O_PATH */
     fd = g_open (path, O_PATH | O_CLOEXEC | O_RDONLY | O_NOFOLLOW);
+#endif
 
   errsv = errno;
 
diff -u -r glib-2.62.6-orig/gio/gunixfdmessage.c glib-2.62.6/gio/gunixfdmessage.c
--- glib-2.62.6-orig/gio/gunixfdmessage.c	2020-03-18 13:16:11.000000000 +0000
+++ glib-2.62.6/gio/gunixfdmessage.c	2020-05-31 12:52:34.753378200 +0000
@@ -48,6 +48,10 @@
 #include <fcntl.h>
 #include <errno.h>
 
+#if defined(__sgi)
+# define SCM_RIGHTS 0x01
+#endif
+
 #include "gunixfdmessage.h"
 #include "gunixfdlist.h"
 #include "gnetworking.h"
diff -u -r glib-2.62.6-orig/glib/giounix.c glib-2.62.6/glib/giounix.c
--- glib-2.62.6-orig/glib/giounix.c	2020-03-18 13:16:11.000000000 +0000
+++ glib-2.62.6/glib/giounix.c	2020-05-31 12:52:34.755534520 +0000
@@ -31,7 +31,17 @@
 
 #include "config.h"
 
+#if defined(__sgi)
+/* Needed to get the right declarations for select() before POSIX_SOURCE
+ * define
+ */
+#include <unistd.h>
+#include <sys/types.h>
+#include <bstring.h>
+#include <sys/time.h>
+#else
 #define _POSIX_SOURCE		/* for SSIZE_MAX */
+#endif
 
 #include <sys/types.h>
 #include <sys/stat.h>
diff -u -r glib-2.62.6-orig/glib/glibconfig.h.in glib-2.62.6/glib/glibconfig.h.in
--- glib-2.62.6-orig/glib/glibconfig.h.in	2020-03-18 13:16:11.000000000 +0000
+++ glib-2.62.6/glib/glibconfig.h.in	2020-05-31 12:52:34.757008040 +0000
@@ -16,7 +16,7 @@
  * system printf functions.  This is useful to know, for example,
  * when using glibc's register_printf_function().
  */
-#mesondefine GLIB_USING_SYSTEM_PRINTF
+/*#mesondefine GLIB_USING_SYSTEM_PRINTF*/
 
 #mesondefine GLIB_STATIC_COMPILATION
 #mesondefine GOBJECT_STATIC_COMPILATION
diff -u -r glib-2.62.6-orig/glib/gmain.c glib-2.62.6/glib/gmain.c
--- glib-2.62.6-orig/glib/gmain.c	2020-03-18 13:16:11.000000000 +0000
+++ glib-2.62.6/glib/gmain.c	2020-05-31 12:52:34.764143720 +0000
@@ -2844,7 +2844,11 @@
   struct timespec ts;
   gint result;
 
+#if defined(__sgi)
+  result = clock_gettime (CLOCK_SGI_CYCLE, &ts);
+#else
   result = clock_gettime (CLOCK_MONOTONIC, &ts);
+#endif
 
   if G_UNLIKELY (result != 0)
     g_error ("GLib requires working CLOCK_MONOTONIC");
diff -u -r glib-2.62.6-orig/glib/gnulib/meson.build glib-2.62.6/glib/gnulib/meson.build
--- glib-2.62.6-orig/glib/gnulib/meson.build	2020-03-18 13:16:11.000000000 +0000
+++ glib-2.62.6/glib/gnulib/meson.build	2020-05-31 12:53:40.425962360 +0000
@@ -323,14 +323,16 @@
                          output: 'gnulib_math.h',
                          configuration: math_h_config)
 
-gnulib_sources = ['asnprintf.c', 'printf.c', 'printf-args.c', 'printf-parse.c', 'printf-frexp.c', 'printf-frexpl.c', 'isnand.c', 'isnanl.c', 'vasnprintf.c']
+# We already have all these in libdicl
+gnulib_sources = ['asnprintf.c', 'printf.c', 'printf-args.c', 'printf-parse.c', 'printf-frexp.c', 'printf-frexpl.c', 'isnand.c', 'isnanf.c', 'isnanl.c', 'vasnprintf.c']
+#gnulib_sources = ['printf.c','vasnprintf.c']
 
-if not gl_cv_func_frexp_works
-  gnulib_sources += ['frexp.c']
-endif
-if not gl_cv_func_frexpl_works
-  gnulib_sources += ['frexpl.c']
-endif
+#if not gl_cv_func_frexp_works
+#  gnulib_sources += ['frexp.c']
+#endif
+#if not gl_cv_func_frexpl_works
+#  gnulib_sources += ['frexpl.c']
+#endif
 
 gnulib_sources += other_needed_math_sources
 
diff -u -r glib-2.62.6-orig/glib/gthread-posix.c glib-2.62.6/glib/gthread-posix.c
--- glib-2.62.6-orig/glib/gthread-posix.c	2020-03-18 13:16:11.000000000 +0000
+++ glib-2.62.6/glib/gthread-posix.c	2020-05-31 12:52:34.768641720 +0000
@@ -661,7 +661,7 @@
 
   pthread_condattr_init (&attr);
 
-#ifdef HAVE_PTHREAD_COND_TIMEDWAIT_RELATIVE_NP
+#if defined(HAVE_PTHREAD_COND_TIMEDWAIT_RELATIVE_NP) || defined(__sgi)
 #elif defined (HAVE_PTHREAD_CONDATTR_SETCLOCK) && defined (CLOCK_MONOTONIC)
   if G_UNLIKELY ((status = pthread_condattr_setclock (&attr, CLOCK_MONOTONIC)) != 0)
     g_thread_abort (status, "pthread_condattr_setclock");
@@ -901,6 +901,22 @@
     if ((status = pthread_cond_timedwait_relative_np (g_cond_get_impl (cond), g_mutex_get_impl (mutex), &ts)) == 0)
       return TRUE;
   }
+#elif defined(__sgi)
+  {
+    gint64 now = g_get_monotonic_time();
+    gint64 absendtime;
+
+    if (end_time <= now)
+      return FALSE;
+
+    absendtime = g_get_real_time () + (end_time - now);
+
+    ts.tv_sec = absendtime / 1000000;
+    ts.tv_nsec = (absendtime % 1000000) * 1000;
+
+    if ((status = pthread_cond_timedwait (g_cond_get_impl (cond), g_mutex_get_impl (mutex), &ts)) == 0)
+      return TRUE;
+  }
 #elif defined (HAVE_PTHREAD_CONDATTR_SETCLOCK) && defined (CLOCK_MONOTONIC)
   /* This is the exact check we used during init to set the clock to
    * monotonic, so if we're in this branch, timedwait() will already be
diff -u -r glib-2.62.6-orig/glib/gutils.c glib-2.62.6/glib/gutils.c
--- glib-2.62.6-orig/glib/gutils.c	2020-03-18 13:16:11.000000000 +0000
+++ glib-2.62.6/glib/gutils.c	2020-05-31 12:52:34.772754600 +0000
@@ -992,11 +992,17 @@
       const gsize size_large = (gsize) 256 * 256;
       gchar *tmp;
 
+#if defined(__sgi)
+      max = 0;
+#else
       max = sysconf (_SC_HOST_NAME_MAX);
+#endif
       if (max > 0 && max <= G_MAXSIZE - 1)
         size = (gsize) max + 1;
       else
-#ifdef HOST_NAME_MAX
+#if defined(__sgi)
+        size = MAXHOSTNAMELEN + 1;
+#elif defined(HOST_NAME_MAX)
         size = HOST_NAME_MAX + 1;
 #else
         size = _POSIX_HOST_NAME_MAX + 1;
Only in glib-2.62.6-orig: glib-2.62.6
diff -u -r glib-2.62.6-orig/meson.build glib-2.62.6/meson.build
--- glib-2.62.6-orig/meson.build	2020-03-18 13:16:11.000000000 +0000
+++ glib-2.62.6/meson.build	2020-05-31 12:52:34.776757720 +0000
@@ -449,7 +449,6 @@
   'link',
   'localtime_r',
   'lstat',
-  'mbrtowc',
   'memalign',
   'mmap',
   'newlocale',
@@ -475,12 +474,13 @@
   'valloc',
   'vasprintf',
   'vsnprintf',
-  'wcrtomb',
   'wcslen',
   'wcsnlen',
   'sysctlbyname',
   '_NSGetEnviron',
 ]
+# Not for IRIX  'mbrtowc',
+# Not for IRIX  'wcrtomb',
 
 if glib_conf.has('HAVE_SYS_STATVFS_H')
   functions += ['statvfs']
diff -u -r glib-2.62.6-orig/tests/mapping-test.c glib-2.62.6/tests/mapping-test.c
--- glib-2.62.6-orig/tests/mapping-test.c	2020-03-18 13:16:11.000000000 +0000
+++ glib-2.62.6/tests/mapping-test.c	2020-05-31 12:52:34.778149000 +0000
@@ -208,7 +208,11 @@
   signal (SIGUSR1, handle_usr1);
 #endif
 
+#if defined(__sgi)
+  g_snprintf (pid, sizeof(pid), "%ld", getpid ());
+#else
   g_snprintf (pid, sizeof(pid), "%d", getpid ());
+#endif
   child_argv[0] = argv0;
   child_argv[1] = "mapchild";
   child_argv[2] = pid;
