diff -Naur libatomic_ops-7.6.10.orig/src/atomic_ops.c libatomic_ops-7.6.10/src/atomic_ops.c
--- libatomic_ops-7.6.10.orig/src/atomic_ops.c	2019-03-01 08:59:46.000000000 +0000
+++ libatomic_ops-7.6.10/src/atomic_ops.c	2020-04-08 18:26:03.660217201 +0000
@@ -30,6 +30,7 @@
 
 #if defined(HAVE_CONFIG_H)
 # include "config.h"
+#include <sys/time.h>
 #endif
 
 #if (defined(__hexagon__) || defined(__native_client__)) \
