diff -u -r libsecret-0.19.1-orig/Makefile.am libsecret-0.19.1/Makefile.am
--- libsecret-0.19.1-orig/Makefile.am	2019-09-05 08:29:10.000000000 +0000
+++ libsecret-0.19.1/Makefile.am	2020-11-08 11:52:22.346527360 +0000
@@ -109,7 +109,6 @@
 	-I$(top_srcdir)/build \
 	-DSRCDIR="\"@abs_srcdir@\"" \
 	-DLOCALEDIR=\""$(datadir)/locale"\" \
-	-DWITH_VALGRIND \
 	-DSECRET_COMPILATION \
 	$(LIBGCRYPT_CFLAGS) \
 	$(GLIB_CFLAGS)
diff -u -r libsecret-0.19.1-orig/egg/egg-secure-memory.c libsecret-0.19.1/egg/egg-secure-memory.c
--- libsecret-0.19.1-orig/egg/egg-secure-memory.c	2019-06-17 10:12:51.000000000 +0000
+++ libsecret-0.19.1/egg/egg-secure-memory.c	2020-11-08 11:52:30.722067360 +0000
@@ -26,6 +26,13 @@
  * without linking in any special libraries. ie: the PKCS#11 module.
  */
 
+#if defined(__sgi)
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <errno.h>
+#endif
+
 #include "config.h"
 
 #include "egg-secure-memory.h"
@@ -168,6 +175,9 @@
 	Pool *pool;
 	void *pages, *item;
 	size_t len, i;
+#if defined(__sgi)
+		int mmapfd;
+#endif
 
 	if (!EGG_SECURE_GLOBALS.pool_version ||
 	    strcmp (EGG_SECURE_GLOBALS.pool_version, EGG_SECURE_POOL_VER_STR) != 0) {
@@ -188,7 +198,15 @@
 	/* Create a new pool */
 	if (pool == NULL) {
 		len = getpagesize () * 2;
+#if defined(__sgi)
+		if( (mmapfd = open("/dev/zero", O_RDWR)) == -1 ) {
+			return NULL;
+		}
+		pages = mmap (0, len, PROT_READ | PROT_WRITE, MAP_PRIVATE, mmapfd, 0);
+		close(mmapfd);
+#else
 		pages = mmap (0, len, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANON, -1, 0);
+#endif
 		if (pages == MAP_FAILED)
 			return NULL;
 
@@ -854,6 +872,9 @@
 {
 	void *pages;
 	unsigned long pgsize;
+#if defined(__sgi)
+	int mmapfd;
+#endif
 
 	ASSERT (sz);
 	ASSERT (*sz);
@@ -864,7 +885,16 @@
 	*sz = (*sz + pgsize -1) & ~(pgsize - 1);
 
 #if defined(HAVE_MLOCK)
+
+#if defined(__sgi)
+	if( (mmapfd = open("/dev/zero", O_RDWR)) == -1 ) {
+		return NULL;
+	}
+	pages = mmap (0, *sz, PROT_READ | PROT_WRITE, MAP_PRIVATE, mmapfd, 0);
+	close(mmapfd);
+#else
 	pages = mmap (0, *sz, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANON, -1, 0);
+#endif
 	if (pages == MAP_FAILED) {
 		if (show_warning && egg_secure_warnings)
 			fprintf (stderr, "couldn't map %lu bytes of memory (%s): %s\n",
