diff -Naur netsurf-all-3.9.orig/netsurf/frontends/gtk/gui.c netsurf-all-3.9/netsurf/frontends/gtk/gui.c
--- netsurf-all-3.9.orig/netsurf/frontends/gtk/gui.c	2019-07-17 08:37:00.000000000 +0000
+++ netsurf-all-3.9/netsurf/frontends/gtk/gui.c	2020-09-06 22:49:13.114925126 +0000
@@ -201,7 +201,7 @@
 	}
 
 	/* default path to certificates */
-	nsoption_setnull_charp(ca_path, strdup("/etc/ssl/certs"));
+	nsoption_setnull_charp(ca_path, strdup("/usr/sgug/etc/ssl/certs"));
 
 	if ((nsoption_charp(cookie_file) == NULL) ||
 	    (nsoption_charp(cookie_jar) == NULL) ||
@@ -224,7 +224,9 @@
 	return NSERROR_OK;
 }
 
-
+#ifndef PATH_MAX
+#define PATH_MAX 4096
+#endif
 
 
 /**
